<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">

  <head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  <!-- Enable responsiveness on mobile devices-->
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>
    
      Useful Linux/Unix commands &middot; DNAwesome
    
  </title>

  <!-- CSS -->
  <link rel="stylesheet" href="/public/css/poole.css">
  <link rel="stylesheet" href="/public/css/syntax.css">

  <!-- Icons -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/public/apple-touch-icon-precomposed.png">
  <link rel="shortcut icon" href="/public/favicon.ico">

  <!-- RSS -->
  <link rel="alternate" type="application/rss+xml" title="RSS" href="/atom.xml">
</head>

  <script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-54133406-1', 'auto');
  ga('send', 'pageview');

</script>


  <body>

    <div class="container content">
      <div class="masthead">
        <h3 class="masthead-title">
          <a href="/" title="Home">DNAwesome</a>
          <small></small>

          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/about">About</a></small>
          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/archive">Archive</a></small>
          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/code">Code</a></small>
          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/atom.xml">Feed</a></small>
          
        </h3>
      </div>

      <div class="post">
  <h1 class="post-title">Useful Linux/Unix commands</h1>
  <span class="post-date">24 Aug 2014</span>
  <h2>Do something in a range</h2>

<p>Move files</p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="k">for</span> i in <span class="k">$(</span>seq 10<span class="k">)</span><span class="p">;</span> <span class="k">do</span> mv chr<span class="nv">$i</span> ../genomes/<span class="p">;</span> <span class="k">done</span>
</code></pre></div><div class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="k">for</span> directory in ./*<span class="p">;</span> <span class="k">do</span>
    <span class="k">if</span> <span class="o">[</span> -d <span class="s2">&quot;$directory&quot;</span> <span class="o">]</span><span class="p">;</span> <span class="k">then</span>
        <span class="nb">cd</span> <span class="nv">$directory</span>
        <span class="k">for</span> myfile in <span class="k">$(</span>ls -d *.sra<span class="k">)</span><span class="p">;</span><span class="k">do</span>
            mv <span class="nv">$myfile</span> /home/tstuart/working_data/1000genomes/<span class="nv">$myfile</span>
        <span class="k">done</span>
        <span class="nb">cd</span> ..
    <span class="k">fi</span>
<span class="k">done</span>
</code></pre></div>
<h2>GNU Screen</h2>

<p>Start screen: <code>screen -S [screen name]</code><br>
List screens: <code>screen -list</code><br>
Detach: <code>screen -d</code><br>
Attach: <code>screen -r</code><br>
Close screen: <code>ctr-a-d</code><br>
Kill screen: <code>ctr-a :quit</code>  </p>

<h2>Searching</h2>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">grep <span class="s1">&#39;text&#39;</span> filename
grep <span class="s1">&#39;text&#39;</span> file1 file2 file3
grep <span class="s1">&#39;text1 text2&#39;</span> filename
grep --color <span class="s1">&#39;text&#39;</span> filename
</code></pre></div>
<p>Search all files in a directory, show output in less  </p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">grep -r <span class="s1">&#39;text&#39;</span> /home/usr/ <span class="p">|</span> less
</code></pre></div>
<p>Search for multiple strings  </p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">egrep <span class="s1">&#39;(AT4G40030)|(AT4G40040)|(AT5G10980)&#39;</span> * &gt; h3.3
</code></pre></div>
<h2>Pipe output into file</h2>

<h4>Overwrite contents of file</h4>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">grep <span class="s2">&quot;text&quot;</span> filename.txt &gt; output.txt
</code></pre></div>
<h4>Append to file</h4>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">grep <span class="s2">&quot;text&quot;</span> filename.txt &gt;&gt; output.txt
</code></pre></div>
<h2>Counting</h2>

<p>Count lines in chr1 file</p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">wc -l chr1
</code></pre></div>
<p>Count characters in chr1 file</p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">wc -c chr1
</code></pre></div>
<h2>List</h2>

<p>List and sort  </p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">ls -ls
</code></pre></div>
<p>List and sort by size  </p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">ls -ls -S
</code></pre></div>
<p>Count number of directories in current directory  </p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">ls -l <span class="p">|</span> grep ^d <span class="p">|</span> wc -l
</code></pre></div>
<h2>Sorting</h2>

<p>Sort file by numerical order of first column, save as sorted_list.txt  </p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">sort -nk <span class="m">1</span> list.txt &gt; sorted_list.txt
</code></pre></div>
<p>Sort file by order of first column, then numerical order of second column, save as sorted_list.txt  </p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">sort -k1,1 -nk2,2 list.txt &gt; sorted_list.txt
</code></pre></div>
<p>Sort descending  </p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">sort -rn -k3 file.txt &gt; sorted.txt
</code></pre></div>
<p>overwrite  </p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">sort -rn -k3 -o file.txt file.txt
</code></pre></div>
<h2>Rows</h2>

<p>Delete first line</p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">sed 1d file.txt &gt; headerless_file.txt
</code></pre></div>
<p>Delete lines 1-3 inclusive  </p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">sed 1,3d file.txt &gt; newfile.txt
</code></pre></div>
<p>Delete lines containing string  </p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">sed <span class="s1">&#39;/ATC/d&#39;</span> file.txt &gt; file_mod.txt
</code></pre></div>
<p>Overwrite file  </p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">sed -i.bak 1d file.txt
</code></pre></div>
<h2>Columns</h2>

<p>Write columns to new file. Can also reorder columns.  </p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">awk <span class="s1">&#39;BEGIN {FS=OFS=&quot;\t&quot;} {print $2,$4,$5,$6,$7,$8,$1,$3}&#39;</span> file.txt &gt; outfile.txt
awk <span class="s1">&#39;BEGIN {FS=OFS=&quot;\t&quot;} {print $7,$1,$8,$2,$3,$4,$5,$6}&#39;</span> cmt2_targets.tsv &gt; cmt2_targets_ordered.tsv
</code></pre></div>
<p>Add column  </p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">awk <span class="s1">&#39;{print $0, &quot;cmt2&quot;}&#39;</span> cmt2_targets_ordered.tsv &gt; newfile.tsv
</code></pre></div>
<p>Remove first column from all files  </p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="k">for</span> myfile in <span class="k">$(</span>ls<span class="k">)</span><span class="p">;</span><span class="k">do</span>
    awk <span class="s1">&#39;BEGIN {FS=OFS=&quot;\t&quot;} {$1=&quot;&quot;;sub(&quot;\t&quot;,&quot;&quot;)}1&#39;</span> <span class="nv">$myfile</span> &gt; mod_<span class="nv">$myfile</span>
<span class="k">done</span>
</code></pre></div>
<p>Search only in one column</p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">awk <span class="s1">&#39;{if ($1 == 1) print $2}&#39;</span> p1 &gt; chr1_p1
</code></pre></div>
<h2>Joining files</h2>

<p>Join files with each starting on a new line  </p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="k">for</span> filename in *.fa<span class="p">;</span> <span class="k">do</span>
    cat <span class="s2">&quot;${filename}&quot;</span>
    <span class="nb">echo</span>
<span class="k">done</span> &gt; output.fa
</code></pre></div>
<p>Merge files of the same format</p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">cat p1_* <span class="p">|</span> sort -nk1,1 -k2,2 <span class="p">|</span> uniq &gt; p1
</code></pre></div>
<h2>Comparing Files</h2>

<p>Files should first be sorted</p>

<p>Find lines that are common or different between two files  </p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">comm -i file1 file2 &gt; output.txt
</code></pre></div>
<ul>
<li><code>-i</code> is case-insensitive<br></li>
<li>outputs three columns. First is lines only in file1, second is lines only in file2, third is lines common in both.</li>
</ul>

<h2>Compressing files</h2>

<p>Compress recursively  </p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">tar cvfz slam.tgz slam/
</code></pre></div>
<h2>Download data from SRA</h2>

<h3>Using wget</h3>
<div class="highlight"><pre><code class="language-text" data-lang="text">wget -r --no-parent --reject &quot;index.html*&quot; ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByStudy/sra/SRP/SRP005/SRP005399/
</code></pre></div>
<p>Alternatively, you can download a list of accessions from the SRA website and use that file to call <code>wget</code> for all the accessions.  </p>
<div class="highlight"><pre><code class="language-python" data-lang="python"><span class="kn">from</span> <span class="nn">subprocess</span> <span class="kn">import</span> <span class="n">call</span>


<span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s">&#39;SraAccList.txt&#39;</span><span class="p">,</span> <span class="s">&#39;r&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">accessions</span><span class="p">:</span>
    <span class="k">for</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">accessions</span><span class="p">:</span>
        <span class="n">acc</span> <span class="o">=</span> <span class="n">row</span><span class="o">.</span><span class="n">strip</span><span class="p">(</span><span class="s">&#39;</span><span class="se">\n</span><span class="s">&#39;</span><span class="p">)</span>
        <span class="k">print</span> <span class="s">&quot;Downloading {a}&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">a</span><span class="o">=</span><span class="n">acc</span><span class="p">)</span>
        <span class="n">call</span><span class="p">([</span><span class="s">&quot;wget&quot;</span><span class="p">,</span>
              <span class="s">&quot;-r&quot;</span><span class="p">,</span>
              <span class="s">&quot;--no-parent&quot;</span><span class="p">,</span>
              <span class="s">&quot;ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR492/{acc}/*&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">acc</span><span class="o">=</span><span class="n">acc</span><span class="p">)])</span>
</code></pre></div>
<h4>Split into fastq files</h4>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">fastq-dump --split-3 ./SRP005399/*
</code></pre></div>
<p><code>--split-3</code> causes PE files to be split into <code>*_1.fastq</code> and <code>*_2.fastq</code></p>

<p>bash loop  </p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="k">for</span> directory in ./*<span class="p">;</span> <span class="k">do</span>
    <span class="k">if</span> <span class="o">[</span> -d <span class="s2">&quot;$directory&quot;</span> <span class="o">]</span><span class="p">;</span> <span class="k">then</span>
        <span class="nb">cd</span> <span class="nv">$directory</span>
        <span class="k">for</span> myfile in <span class="k">$(</span>ls -d *.sra<span class="k">)</span><span class="p">;</span><span class="k">do</span>
            fastq-dump --split-3 -v <span class="nv">$myfile</span>
        <span class="k">done</span>
        <span class="nb">cd</span> ..
    <span class="k">fi</span>
<span class="k">done</span>
</code></pre></div>
<p>Python function  </p>
<div class="highlight"><pre><code class="language-python" data-lang="python"><span class="kn">import</span> <span class="nn">os</span>
<span class="kn">from</span> <span class="nn">subprocess</span> <span class="kn">import</span> <span class="n">call</span>


<span class="k">def</span> <span class="nf">fastqSplit</span><span class="p">():</span>
    <span class="k">for</span> <span class="n">filename</span> <span class="ow">in</span> <span class="n">os</span><span class="o">.</span><span class="n">listdir</span><span class="p">(</span><span class="s">&#39;.&#39;</span><span class="p">):</span>
        <span class="k">if</span> <span class="n">filename</span><span class="o">.</span><span class="n">endswith</span><span class="p">(</span><span class="s">&#39;.sra&#39;</span><span class="p">):</span>
            <span class="k">print</span> <span class="s">&#39;processing {n}&#39;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">n</span><span class="o">=</span><span class="n">filename</span><span class="p">)</span>
            <span class="n">call</span><span class="p">([</span><span class="s">&#39;fastq-dump&#39;</span><span class="p">,</span> <span class="s">&#39;--split-3&#39;</span><span class="p">,</span> <span class="s">&#39;-v&#39;</span><span class="p">,</span> <span class="n">filename</span><span class="p">])</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">pass</span>
</code></pre></div>
<h3>Using SRA toolkit</h3>

<p>SRA toolkit doesn&#39;t seem to work well for downloading bulk data. <code>wget</code> is a much better option as it allows download of whole folders, gives more descriptive output as it goes. SRA often fails and sometimes gives no error. Download with <code>wget</code> and use <code>fastq-split</code> to get fastq files.  </p>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">fastq-dump SRR534224 <span class="p">&amp;</span>
</code></pre></div>
<h2>RNA-seq</h2>

<h3>Mapping</h3>
<div class="highlight"><pre><code class="language-text" data-lang="text">tophat -p8 -G /home/lister/working_data/data/genomes/annotations/tair10/TAIR10_gene_TE_illumina.gtf --transcriptome-index=transcript_index/tair10 /home/lister/working_data/data/genomes/bowtie2_indexes/tair9 SRR501604.fastq,SRR501605.fastq
</code></pre></div>
<ul>
<li><code>--transcriptome-index=transcript_index/tair10</code> creates transcriptome index file saved in transcript_index/ with name <code>tair10.*</code>. This can be reused if mapping to the same transcriptome (ie. gff file)<br></li>
<li><code>-p8</code> uses 8 cores<br></li>
<li><code>-G</code> specifies gff file. Optional<br></li>
<li>If using PE reads check manual<br></li>
</ul>

<h2>Bedtools</h2>

<p><a href="http://quinlanlab.org/tutorials/cshl2013/bedtools.html">Good tutorial from the Quinlan lab</a>  </p>

<p>Need to order files so that chromosome, start, stop and first three columns (BED format). Also need to remove header. Can do these steps with <code>awk</code> and <code>sed 1d</code>.  </p>

<h3>Merge</h3>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">bedtools merge -i dmrs_ddc.tsv -d <span class="m">100</span> &gt; dmrs_ddc_merged.tsv
</code></pre></div>
<ul>
<li><code>-d</code> is distance between coordinates that will still be merged<br></li>
<li><code>-i</code> is input file<br></li>
</ul>

<h3>Intersect</h3>
<div class="highlight"><pre><code class="language-bash" data-lang="bash">bedtools intersect -a tair10_tes.txt -b dmrs_ddc_merged.tsv -wa -f 0.50 &gt; ddc_targets.tsv
</code></pre></div>
<ul>
<li><code>-a</code> is file a<br></li>
<li><code>-b</code> is file b<br></li>
<li><code>-wa</code> is write a<br></li>
<li><code>-f</code> is minimum overlap percentage<br></li>
</ul>

<h2>Bowtie</h2>

<h3>Mapping PE data</h3>
<div class="highlight"><pre><code class="language-text" data-lang="text">bowtie2 -p8 --local --fr -q -R 5 -N 1 -x [path/to/bowtie2/index] -X 1000 -1 [mate1.fq] -2 [mate2.fq] -S [alignment.sam] --un-conc ./discordant/
</code></pre></div>
<ul>
<li><code>-p</code> is number of processors<br></li>
<li><code>--local</code> allows soft-clipping of reads to improve alignment<br></li>
<li><code>-q</code> specifies that reads are in fastq format<br></li>
<li><code>-x</code> gives path to index<br></li>
<li><code>-S</code> gives name of output file<br></li>
<li><code>-R</code> is number of times bowtie will try to &#39;re-seed&#39; repetitive seeds. Default 2<br></li>
<li><code>-N</code> is number of mismatches allowed in seed. 0 or 1.<br></li>
<li><code>--no-mixed</code> tells bowtie to find alignments only when both pairs can be aligned.<br></li>
<li><code>--fr</code> means mate pairs are ordered in forward then reverse orientation. Can do <code>--ff</code>, <code>--rf</code>.<br></li>
<li><code>-X</code> specifies maximum insert size. Default 500.<br></li>
<li><code>-I</code> specifies minimum insert size. Default 0 (no minimum).<br></li>
<li><code>--un-conc</code> specifies path and file to write discordant alignments to. Note that these are just the <code>fastq</code> reads, not alignments.</li>
</ul>

<h2>Samtools</h2>

<p>samtools can read from a stream, so can pipe output in from other tools (eg bowtie to get <code>.bam</code> output) or other samtools commands.</p>

<h3>Convert from <code>sam</code> to <code>bam</code></h3>
<div class="highlight"><pre><code class="language-text" data-lang="text">samtools view -bS file.sam &gt; file.bam
</code></pre></div>
<h3>Sort bamfile</h3>
<div class="highlight"><pre><code class="language-text" data-lang="text">samtools sort file.bam sorted
</code></pre></div>
<ul>
<li>outputs <code>sorted.bam</code> file</li>
</ul>

<h3>Samblaster</h3>

<p>Extracts reads from <code>.sam</code> alignment files</p>

<h4>Get discordant reads</h4>
<div class="highlight"><pre><code class="language-text" data-lang="text">samblaster -e -d file.sam | samtools view -bS - &gt; file.bam
</code></pre></div>
<ul>
<li>pipe output directly into <code>samtools view</code> to save as <code>.bam</code> file</li>
</ul>

</div>

<div class="related">
  <h2>Related Posts</h2>
  <ul class="related-posts">
    
      <li>
        <h3>
          <a href="//2014/08/25/UsefulPython/">
            Useful python code
            <small>25 Aug 2014</small>
          </a>
        </h3>
      </li>
    
      <li>
        <h3>
          <a href="//2014/03/04/highly-multiplexed-subcellular-rna-sequencing-in-situ/">
            Highly Multiplexed Subcellular RNA Sequencing in Situ
            <small>04 Mar 2014</small>
          </a>
        </h3>
      </li>
    
  </ul>
</div>


      <!-- <div class="footer">
        <p>
          &copy; 2014. All rights reserved.
        </p>
      </div> -->
    </div>

  </body>
</html>
