<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">

  <head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  <!-- Enable responsiveness on mobile devices-->
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>
    
      Tim Stuart &middot; 
    
  </title>

  <!-- CSS -->
  <link rel="stylesheet" href="/public/css/poole.css">
  <link rel="stylesheet" href="/public/css/syntax.css">

  <!-- Icons -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/public/apple-touch-icon-precomposed.png">
  <link rel="shortcut icon" href="/public/favicon.ico">

  <!-- RSS -->
  <link rel="alternate" type="application/rss+xml" title="RSS" href="/atom.xml">
</head>

  <script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-54133406-1', 'auto');
  ga('send', 'pageview');

</script>


  <body>

    <div class="container content">
      <div class="masthead">
        <h3 class="masthead-title">
          <a href="/" title="Home">Tim Stuart</a>
          <small></small>

          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/about">About</a></small>
          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/archive">Archive</a></small>
          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/publications">Publications</a></small>
          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/atom.xml">Feed</a></small>
          
        </h3>
      </div>

      <div class="posts">
  
   <div class="post-preview">
   <h2><a href="/2017/09/14/useful-bioinformatics/">Useful bioinformatics</a></h2>
   <span class="post-date">September 14, 2017</span>
   
     <h2 id="trim-reads">Trim reads</h2>

<div class="language-bash highlighter-rouge"><pre class="highlight"><code>cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC -m 30 -o output.fq.gz input.fq.gz
</code></pre>
</div>


     <a href="/2017/09/14/useful-bioinformatics/">read more</a>
   
   </div>
   <hr>
   
   <div class="post-preview">
   <h2><a href="/2017/04/09/using-python-decorators/">Using python decorators</a></h2>
   <span class="post-date">April 09, 2017</span>
   
     <p>Yesterday I wrote my first python decorator. Decorators have always seemed a bit mysterious to me, but having finally written one I can see a bit better how they work. This is the decorator I wrote:</p>


     <a href="/2017/04/09/using-python-decorators/">read more</a>
   
   </div>
   <hr>
   
   <div class="post-preview">
   <h2><a href="/2017/04/03/comp-notebook/">Create a computational lab notebook with bookdown</a></h2>
   <span class="post-date">April 03, 2017</span>
   
     <p>Every data analysis I do now is kept in an <a href="http://rmarkdown.rstudio.com">R Markdown</a> document. These are great for mixing code with explanatory text, and you can run code in many languages not just R. Whenever I finished working on something, I would compile the R Markdown document into a self-contained html report and save is somewhere, usually with a descriptive filename like “coverage_genes” or “col_vs_cvi”.</p>


     <a href="/2017/04/03/comp-notebook/">read more</a>
   
   </div>
   <hr>
   
   <div class="post-preview">
   <h2><a href="/2017/03/03/installing-magic/">Installing Magic</a></h2>
   <span class="post-date">March 03, 2017</span>
   
     <h1 id="installing-magic">Installing magic</h1>

<p>Recently a method for imputing single cell gene expression matricies
was posted
on <a href="http://biorxiv.org/content/early/2017/02/25/111591">biorxiv</a> by
David van Dijk et al., called magic (Markov Affinity-based Graph
Imputation of Cells). I’ve been analysing single cell RNA-seq data
recently, and this method looks like it could be useful when trying to
find co-transcriptional networks, as single cell data suffers from
dropout which makes finding co-transcriptional networks hard.</p>


     <a href="/2017/03/03/installing-magic/">read more</a>
   
   </div>
   <hr>
   
   <div class="post-preview">
   <h2><a href="/2016/09/22/r_demo/">R demo</a></h2>
   <span class="post-date">September 22, 2016</span>
   
     <p><strong>Table of Contents</strong></p>

<ul>
  <li><a href="#getting-started">Getting started</a></li>
  <li><a href="#importing-data">Importing Data</a></li>
  <li><a href="#tidying-data">Tidying data</a></li>
  <li><a href="#manipulating-data">Manipulating data</a></li>
  <li><a href="#put-it-all-together">Put it all together</a></li>
  <li><a href="#plotting">Plotting</a></li>
  <li><a href="#reporting">Reporting</a></li>
  <li><a href="#further-reading">Further reading</a></li>
  <li><a href="#session-info">Session info</a></li>
</ul>

<h2 id="getting-started">Getting started</h2>

<p>Clone the repo if you haven’t already:</p>

<div class="language-bash highlighter-rouge"><pre class="highlight"><code>git clone https://github.com/timoast/dac.git
</code></pre>
</div>

<p>Install <a href="https://www.rstudio.com/">RStudio</a>.</p>

<p>Install the following packages:</p>


     <a href="/2016/09/22/r_demo/">read more</a>
   
   </div>
   <hr>
   
   <div class="post-preview">
   <h2><a href="/2016/07/12/smRNA-analysis/">smRNA analysis notes</a></h2>
   <span class="post-date">July 12, 2016</span>
   
     <p>I recently analysed some smRNA data for a paper I’m working on. These are my analysis notes.</p>

<p>I used previously published data for Brachypodium, from this paper:</p>

<p>Garvin DF, Schmutz J, Rokhsar D, Bevan MW, Barry K, Lucas S, et al. Genome sequencing and analysis of the model grass Brachypodium distachyon. Nature. 2010;463: 763–768. doi:10.1038/nature08747</p>

<p>First step is to download the data:</p>

<div class="language-bash highlighter-rouge"><pre class="highlight"><code><span class="gp">$ </span>wget ftp://ftp-trace.ncbi.nlm.nih.gov//sra/sra-instant/reads/ByStudy/sra/SRP/SRP001/SRP001895/SRR035616/SRR035616.sra
<span class="gp">$ </span>fastq-dump SRR035616.sra
<span class="gp">$ </span>pigz SRR035616.fastq
</code></pre>
</div>


     <a href="/2016/07/12/smRNA-analysis/">read more</a>
   
   </div>
   <hr>
   
   <div class="post-preview">
   <h2><a href="/2016/03/29/bsseeker2/">WGBS Analysis notes -- BS-seeker2</a></h2>
   <span class="post-date">March 29, 2016</span>
   
     <h2 id="whole-genome-bisulfite-sequencing-analysis-notes-for-bs-seeker2">Whole genome bisulfite sequencing analysis notes for BS-seeker2.</h2>

<h2 id="step-1-trim-adapters-and-low-quality-bases">Step 1: Trim adapters and low quality bases</h2>
<p>Illumina TruSeq adapter sequence: <code class="highlighter-rouge">GATCGGAAGAGCACACGTCTGAACTCCAGTCAC</code></p>

<div class="highlighter-rouge"><pre class="highlight"><code>cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC -m 50 reads.fastq \
| seqtk trimfq -l 50 - \
| pigz &gt; filtered_reads.fq.gz
</code></pre>
</div>


     <a href="/2016/03/29/bsseeker2/">read more</a>
   
   </div>
   <hr>
   
   <div class="post-preview">
   <h2><a href="/2016/03/01/biorxiv/">bioRxiv</a></h2>
   <span class="post-date">March 01, 2016</span>
   
     <h1 id="a-look-at-biorxiv-preprints">A look at bioRxiv preprints</h1>
<p>Tim Stuart<br />
2 March 2016</p>

<p>After posting a my first <a href="http://biorxiv.org/content/early/2016/02/21/039511">preprint</a> to <a href="http://biorxiv.org/">bioRxiv</a> a few weeks ago, I have been periodically checking the number of views and PDF downloads. I became interested to see how many downloads or views the preprints on bioRxiv typically get, but this type of information isn’t actually available. What are the all-time top bioRxiv preprints? How many people are  reading bioRxiv preprints on average? No-one knows! Altmetric must track this data, as it will tell you how a particular preprint ranks in relation to others, but that data hasn’t been made publicly available (as far as I can tell).</p>


     <a href="/2016/03/01/biorxiv/">read more</a>
   
   </div>
   <hr>
   
   <div class="post-preview">
   <h2><a href="/2016/02/23/Papers/">A paper a day</a></h2>
   <span class="post-date">February 23, 2016</span>
   
     <p>In an effort to read more papers this year, I’m going to read a paper (or something paper-like) each day for the remainder of the year, and post each 
paper below as I go.</p>

<h3 id="dec-31">Dec 31</h3>

<p>Galanter JM, Gignoux CR, Oh SS, Torgerson D, Pino-Yanes M, Thakur N, et al. Differential methylation between ethnic sub-groups reflects the effect of genetic ancestry and environmental exposures. eLife. <a href="http://dx.doi.org/10.7554/eLife.20532">doi:10.7554/eLife.20532</a></p>


     <a href="/2016/02/23/Papers/">read more</a>
   
   </div>
   <hr>
   
   <div class="post-preview">
   <h2><a href="/2016/02/18/Lorne/">Lorne Genome 2016 presentation</a></h2>
   <span class="post-date">February 18, 2016</span>
   
     <p>This week I presented some recent work at the Lorne Genome conference. It was my first time at Lorne and it’s a fantastic meeting with lots of interesting talks. It was great to see so many people getting interested in transposon biology too!</p>

<p>Here are a few links to the work I was presenting:</p>

<p>TEPID TE presence/absence variant discovery software:</p>

<p><a href="https://github.com/ListerLab/TEPID">https://github.com/ListerLab/TEPID</a></p>

<p>Poster:</p>

<p><a href="https://dx.doi.org/10.6084/m9.figshare.2082757.v1">https://dx.doi.org/10.6084/m9.figshare.2082757.v1</a></p>

<p>Manuscript:</p>

<p><a href="http://biorxiv.org/content/early/2016/02/11/039511">http://dx.doi.org/10.1101/039511</a></p>


     <a href="/2016/02/18/Lorne/">read more</a>
   
   </div>
   <hr>
   
</div>

<div class="pagination">
  
    <a class="pagination-item older" href="/page6">Older</a>
  
  
    
      <a class="pagination-item newer" href="/page4">Newer</a>
    
  
</div>


      <!-- <div class="footer">
        <p>
          &copy; 2017. All rights reserved.
        </p>
      </div> -->
    </div>

  </body>
</html>
