<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">

  <head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  <!-- Enable responsiveness on mobile devices-->
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>
    
      R demo &middot; Tim Stuart
    
  </title>

  <!-- CSS -->
  <link rel="stylesheet" href="/public/css/poole.css">
  <link rel="stylesheet" href="/public/css/syntax.css">

  <!-- Icons -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/public/apple-touch-icon-precomposed.png">
  <link rel="shortcut icon" href="/public/favicon.ico">

  <!-- libs -->
  <link href="/libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
  <script src="/libs/pagedtable-1.1/js/pagedtable.js"></script>
  
  <!-- RSS -->
  <link rel="alternate" type="application/rss+xml" title="RSS" href="/atom.xml">
</head>

  <script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-54133406-1', 'auto');
  ga('send', 'pageview');

</script>


  <body>

    <div class="container content">
      <div class="masthead">
        <h3 class="masthead-title">
          <a href="/" title="Home">Tim Stuart</a>
          <small></small>

          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/about">About</a></small>
          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/archive">Archive</a></small>
          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/publications">Publications</a></small>
          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/atom.xml">Feed</a></small>
          
        </h3>
      </div>

      <div class="post">
  <h1 class="post-title">R demo</h1>
  <span class="post-date">22 Sep 2016</span>
  <p><strong>Table of Contents</strong></p>

<ul>
  <li><a href="#getting-started">Getting started</a></li>
  <li><a href="#importing-data">Importing Data</a></li>
  <li><a href="#tidying-data">Tidying data</a></li>
  <li><a href="#manipulating-data">Manipulating data</a></li>
  <li><a href="#put-it-all-together">Put it all together</a></li>
  <li><a href="#plotting">Plotting</a></li>
  <li><a href="#reporting">Reporting</a></li>
  <li><a href="#further-reading">Further reading</a></li>
  <li><a href="#session-info">Session info</a></li>
</ul>

<h2 id="getting-started">Getting started</h2>

<p>Clone the repo if you haven’t already:</p>

<div class="language-bash highlighter-rouge"><pre class="highlight"><code>git clone https://github.com/timoast/dac.git
</code></pre>
</div>

<p>Install <a href="https://www.rstudio.com/">RStudio</a>.</p>

<p>Install the following packages:</p>

<!--break-->

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">install.packages</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"readr"</span><span class="p">,</span><span class="w"> </span><span class="s2">"tidyr"</span><span class="p">,</span><span class="w"> </span><span class="s2">"data.table"</span><span class="p">,</span><span class="w"> </span><span class="s2">"dplyr"</span><span class="p">,</span><span class="w"> </span><span class="s2">"ggplot2"</span><span class="p">))</span><span class="w">
</span></code></pre>
</div>

<p>All the following code will assume you are in the directory that holds this file (<code class="highlighter-rouge">r_demo.rmd</code>) in the github repo (ie <code class="highlighter-rouge">dac/code/</code>).</p>

<h2 id="importing-data">Importing Data</h2>

<h3 id="the-readr-way">The <code class="highlighter-rouge">readr</code> way</h3>

<p><a href="https://blog.rstudio.org/2016/08/05/readr-1-0-0/">Some docs</a></p>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">readr</span><span class="p">)</span><span class="w">

</span><span class="n">read_tsv</span><span class="p">(</span><span class="s2">"../data/coverages.tsv"</span><span class="p">,</span><span class="w"> </span><span class="n">col_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"Accession"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Coverage"</span><span class="p">))</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## Parsed with column specification:
## cols(
##   Accession = col_character(),
##   Coverage = col_integer()
## )
</code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## # A tibble: 216 × 2
##    Accession Coverage
##        &lt;chr&gt;    &lt;int&gt;
## 1       Aa-0       41
## 2      Abd-0       67
## 3       Ag-0       47
## 4       Ak-1       38
## 5     Alst-1       42
## 6    Altai-5       42
## 7     Amel-1       60
## 8       An-1       39
## 9      Ang-0       48
## 10  Anholt-1       53
## # ... with 206 more rows
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="c1"># also works for gzipped files
</span><span class="n">read_tsv</span><span class="p">(</span><span class="s2">"../data/mc_data.bed.gz"</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## Parsed with column specification:
## cols(
##   coords = col_character(),
##   ins0 = col_double(),
##   ins1 = col_double()
## )
</code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## # A tibble: 999 × 3
##              coords        ins0        ins1
##               &lt;chr&gt;       &lt;dbl&gt;       &lt;dbl&gt;
## 1    chr1,8668,8746 0.000000000 0.005366726
## 2  chr1,17024,18924 0.005193408 0.004104893
## 3  chr1,18331,18642 0.003720434 0.008686003
## 4  chr1,23022,23145 0.000000000 0.004184100
## 5  chr1,55676,56576 0.004172405 0.012852597
## 6  chr1,69022,69135 0.008230453 0.004237288
## 7  chr1,69025,69143 0.003484013 0.003608475
## 8  chr1,69025,69148 0.002777778 0.007604563
## 9  chr1,69028,69133 0.005494505 0.000000000
## 10 chr1,76844,77500 0.003282397 0.003786529
## # ... with 989 more rows
</code></pre>
</div>

<h3 id="the-datatable-way">The <code class="highlighter-rouge">data.table</code> way</h3>

<p><a href="https://github.com/Rdatatable/data.table/wiki">Some docs</a></p>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span><span class="w">

</span><span class="n">fread</span><span class="p">(</span><span class="s2">"../data/coverages.tsv"</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">col.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"Accession"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Coverage"</span><span class="p">))</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>##      Accession Coverage
##   1:      Aa-0       41
##   2:     Abd-0       67
##   3:      Ag-0       47
##   4:      Ak-1       38
##   5:    Alst-1       42
##  ---                   
## 212:      Ws-2       52
## 213:      Wt-5       29
## 214:      Yo-0       78
## 215:     Zal-1       41
## 216:     Zdr-1       19
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">fread</span><span class="p">(</span><span class="s2">"gzip -dc ../data/mc_data.bed.gz"</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>##                      coords        ins0        ins1
##   1:         chr1,8668,8746 0.000000000 0.005366726
##   2:       chr1,17024,18924 0.005193408 0.004104893
##   3:       chr1,18331,18642 0.003720434 0.008686003
##   4:       chr1,23022,23145 0.000000000 0.004184100
##   5:       chr1,55676,56576 0.004172405 0.012852597
##  ---                                               
## 995: chr1,12566243,12568303 0.264952130 0.076252087
## 996: chr1,12581911,12582918 0.251092213 0.248537692
## 997: chr1,12584274,12584300 0.082298985 0.096469517
## 998: chr1,12594543,12594591 0.333835719 0.404032186
## 999: chr1,12612939,12613280 0.357346865 0.024760132
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">fread</span><span class="p">(</span><span class="s2">"seq 25 | xargs -n 5 echo"</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>##    V1 V2 V3 V4 V5
## 1:  1  2  3  4  5
## 2:  6  7  8  9 10
## 3: 11 12 13 14 15
## 4: 16 17 18 19 20
## 5: 21 22 23 24 25
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">fread</span><span class="p">(</span><span class="s2">"curl https://raw.githubusercontent.com/timoast/Arabidopsis-TE-variants/master/ProcessedData/coverages.tsv"</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>##          V1 V2
##   1:   Aa-0 41
##   2:  Abd-0 67
##   3:   Ag-0 47
##   4:   Ak-1 38
##   5: Alst-1 42
##  ---          
## 212:   Ws-2 52
## 213:   Wt-5 29
## 214:   Yo-0 78
## 215:  Zal-1 41
## 216:  Zdr-1 19
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">fread</span><span class="p">(</span><span class="s2">"bedtools intersect -a ../data/a.bed -b ../data/b.bed"</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>##      V1 V2  V3
## 1: chr1 20 100
## 2: chr1 50 100
## 3: chr1 20  50
</code></pre>
</div>

<p><code class="highlighter-rouge">data.table</code> has lots of other useful functions, see <a href="https://www.datacamp.com/courses/data-table-data-manipulation-r-tutorial">datacamp course</a></p>

<h3 id="brief-tangent--using-fread-to-eliminate-intermediate-files">Brief tangent – using fread to eliminate intermediate files</h3>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">centromere_distance</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">fread</span><span class="p">(</span><span class="s2">"bedtools closest -d -a ../data/c_dmrs.tsv.gz -b ../data/centromere_positions.txt"</span><span class="p">)</span><span class="w">

</span><span class="n">hist</span><span class="p">(</span><span class="n">centromere_distance</span><span class="o">$</span><span class="n">V</span><span class="m">7</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="m">1000</span><span class="p">,</span><span class="w"> </span><span class="n">breaks</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"blue"</span><span class="p">,</span><span class="w">
     </span><span class="n">main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"C-DMR distances to centromere"</span><span class="p">,</span><span class="w">
     </span><span class="n">xlab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Distance (kb)"</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<p><img src="/figure/./r_demo/plotting_example-1.png" alt="plot of chunk plotting_example" /></p>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">densities</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">hist</span><span class="p">(</span><span class="n">centromere_distance</span><span class="o">$</span><span class="n">V</span><span class="m">7</span><span class="p">,</span><span class="w"> </span><span class="n">breaks</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">500</span><span class="p">,</span><span class="w"> </span><span class="n">plot</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="p">)</span><span class="w">
</span><span class="n">image</span><span class="p">(</span><span class="n">as.matrix</span><span class="p">(</span><span class="n">densities</span><span class="o">$</span><span class="n">counts</span><span class="p">))</span><span class="w">
</span></code></pre>
</div>

<p><img src="/figure/./r_demo/plotting_example-2.png" alt="plot of chunk plotting_example" /></p>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">RColorBrewer</span><span class="p">)</span><span class="w">
</span><span class="n">colors</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">colorRampPalette</span><span class="p">(</span><span class="n">brewer.pal</span><span class="p">(</span><span class="m">9</span><span class="p">,</span><span class="s2">"RdYlBu"</span><span class="p">))(</span><span class="m">100</span><span class="p">)</span><span class="w">
</span><span class="n">image</span><span class="p">(</span><span class="n">as.matrix</span><span class="p">(</span><span class="n">densities</span><span class="o">$</span><span class="n">counts</span><span class="p">),</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">rev</span><span class="p">(</span><span class="n">colors</span><span class="p">))</span><span class="w">
</span></code></pre>
</div>

<p><img src="/figure/./r_demo/plotting_example-3.png" alt="plot of chunk plotting_example" /></p>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="c1"># create a scale bar
</span><span class="n">image</span><span class="p">(</span><span class="n">matrix</span><span class="p">(</span><span class="n">seq</span><span class="p">(</span><span class="m">100</span><span class="p">)),</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">rev</span><span class="p">(</span><span class="n">colors</span><span class="p">))</span><span class="w">
</span></code></pre>
</div>

<p><img src="/figure/./r_demo/plotting_example-4.png" alt="plot of chunk plotting_example" /></p>

<h2 id="tidying-data">Tidying data</h2>

<p>Some background: <a href="https://www.jstatsoft.org/article/view/v059i10">Tidy data, Wickham 2014</a></p>

<p>Variables go in the columns, observations go in the rows.</p>

<p>Four main operations in tidying data:</p>

<ul>
  <li>Move observations from the column names to rows</li>
  <li>Move variables from rows to columns</li>
  <li>Separate multiple variables into multiple columns</li>
  <li>Gather single variable from many columns into one column</li>
</ul>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">tidyr</span><span class="p">)</span><span class="w">

</span><span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read_tsv</span><span class="p">(</span><span class="s2">"../data/experiment.tsv"</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## Parsed with column specification:
## cols(
##   position = col_integer(),
##   experiment_1 = col_double(),
##   experiment_2 = col_double(),
##   experiment_3 = col_double(),
##   experiment_4 = col_double()
## )
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">data</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## # A tibble: 500 × 5
##    position experiment_1 experiment_2 experiment_3 experiment_4
##       &lt;int&gt;        &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;
## 1         1    0.2947252    0.0000000   0.69056274    0.2240412
## 2         2    0.2624428    0.1944636   0.02995589    0.2823043
## 3         3    0.2227268    0.1696682   0.64657079    0.3024177
## 4         4    0.1863447    0.3532347   0.67337368    0.2075265
## 5         5    0.2532761    0.7560800   0.52507202    0.2844748
## 6         6    0.1269957    0.0000000   0.69156885    0.3399855
## 7         7    0.1476328    0.3013359   0.19931927    0.4005465
## 8         8    0.2202941    0.1325935   0.31909674    0.3487259
## 9         9    0.1886488    0.4738528   0.69180550    0.4154950
## 10       10    0.1971425    0.4245025   0.99153173    0.3800341
## # ... with 490 more rows
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">tidy_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gather</span><span class="p">(</span><span class="n">data</span><span class="p">,</span><span class="w"> </span><span class="n">experiment</span><span class="p">,</span><span class="w"> </span><span class="n">mC</span><span class="p">,</span><span class="w"> </span><span class="n">experiment_1</span><span class="o">:</span><span class="n">experiment_4</span><span class="p">)</span><span class="w">
</span><span class="n">tidy_data</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## # A tibble: 2,000 × 3
##    position   experiment        mC
##       &lt;int&gt;        &lt;chr&gt;     &lt;dbl&gt;
## 1         1 experiment_1 0.2947252
## 2         2 experiment_1 0.2624428
## 3         3 experiment_1 0.2227268
## 4         4 experiment_1 0.1863447
## 5         5 experiment_1 0.2532761
## 6         6 experiment_1 0.1269957
## 7         7 experiment_1 0.1476328
## 8         8 experiment_1 0.2202941
## 9         9 experiment_1 0.1886488
## 10       10 experiment_1 0.1971425
## # ... with 1,990 more rows
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">spread</span><span class="p">(</span><span class="n">tidy_data</span><span class="p">,</span><span class="w"> </span><span class="n">experiment</span><span class="p">,</span><span class="w"> </span><span class="n">mC</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## # A tibble: 500 × 5
##    position experiment_1 experiment_2 experiment_3 experiment_4
## *     &lt;int&gt;        &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;
## 1         1    0.2947252    0.0000000   0.69056274    0.2240412
## 2         2    0.2624428    0.1944636   0.02995589    0.2823043
## 3         3    0.2227268    0.1696682   0.64657079    0.3024177
## 4         4    0.1863447    0.3532347   0.67337368    0.2075265
## 5         5    0.2532761    0.7560800   0.52507202    0.2844748
## 6         6    0.1269957    0.0000000   0.69156885    0.3399855
## 7         7    0.1476328    0.3013359   0.19931927    0.4005465
## 8         8    0.2202941    0.1325935   0.31909674    0.3487259
## 9         9    0.1886488    0.4738528   0.69180550    0.4154950
## 10       10    0.1971425    0.4245025   0.99153173    0.3800341
## # ... with 490 more rows
</code></pre>
</div>

<p>Unite and separate columns</p>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">mc</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read_tsv</span><span class="p">(</span><span class="s2">"../data/mc_data.bed.gz"</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## Parsed with column specification:
## cols(
##   coords = col_character(),
##   ins0 = col_double(),
##   ins1 = col_double()
## )
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">head</span><span class="p">(</span><span class="n">mc</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## # A tibble: 6 × 3
##             coords        ins0        ins1
##              &lt;chr&gt;       &lt;dbl&gt;       &lt;dbl&gt;
## 1   chr1,8668,8746 0.000000000 0.005366726
## 2 chr1,17024,18924 0.005193408 0.004104893
## 3 chr1,18331,18642 0.003720434 0.008686003
## 4 chr1,23022,23145 0.000000000 0.004184100
## 5 chr1,55676,56576 0.004172405 0.012852597
## 6 chr1,69022,69135 0.008230453 0.004237288
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="c1"># first column has three variables: chromsome, start, end
</span><span class="n">new_mc</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">separate</span><span class="p">(</span><span class="n">mc</span><span class="p">,</span><span class="w"> </span><span class="n">coords</span><span class="p">,</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"chr"</span><span class="p">,</span><span class="w"> </span><span class="s2">"start"</span><span class="p">,</span><span class="w"> </span><span class="s2">"end"</span><span class="p">))</span><span class="w">
</span><span class="n">new_mc</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## # A tibble: 999 × 5
##      chr start   end        ins0        ins1
## *  &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;       &lt;dbl&gt;       &lt;dbl&gt;
## 1   chr1  8668  8746 0.000000000 0.005366726
## 2   chr1 17024 18924 0.005193408 0.004104893
## 3   chr1 18331 18642 0.003720434 0.008686003
## 4   chr1 23022 23145 0.000000000 0.004184100
## 5   chr1 55676 56576 0.004172405 0.012852597
## 6   chr1 69022 69135 0.008230453 0.004237288
## 7   chr1 69025 69143 0.003484013 0.003608475
## 8   chr1 69025 69148 0.002777778 0.007604563
## 9   chr1 69028 69133 0.005494505 0.000000000
## 10  chr1 76844 77500 0.003282397 0.003786529
## # ... with 989 more rows
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">unite</span><span class="p">(</span><span class="n">new_mc</span><span class="p">,</span><span class="w"> </span><span class="n">coords</span><span class="p">,</span><span class="w"> </span><span class="n">chr</span><span class="p">,</span><span class="w"> </span><span class="n">start</span><span class="p">,</span><span class="w"> </span><span class="n">end</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">","</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## # A tibble: 999 × 3
##              coords        ins0        ins1
## *             &lt;chr&gt;       &lt;dbl&gt;       &lt;dbl&gt;
## 1    chr1,8668,8746 0.000000000 0.005366726
## 2  chr1,17024,18924 0.005193408 0.004104893
## 3  chr1,18331,18642 0.003720434 0.008686003
## 4  chr1,23022,23145 0.000000000 0.004184100
## 5  chr1,55676,56576 0.004172405 0.012852597
## 6  chr1,69022,69135 0.008230453 0.004237288
## 7  chr1,69025,69143 0.003484013 0.003608475
## 8  chr1,69025,69148 0.002777778 0.007604563
## 9  chr1,69028,69133 0.005494505 0.000000000
## 10 chr1,76844,77500 0.003282397 0.003786529
## # ... with 989 more rows
</code></pre>
</div>

<h2 id="manipulating-data">Manipulating data</h2>

<p>The <a href="https://cran.rstudio.com/web/packages/dplyr/vignettes/introduction.html">dplyr</a> package:</p>

<ul>
  <li><code class="highlighter-rouge">filter</code> rows</li>
  <li><code class="highlighter-rouge">select</code> columns</li>
  <li><code class="highlighter-rouge">mutate</code> columns</li>
  <li><code class="highlighter-rouge">arrange</code> rows</li>
  <li><code class="highlighter-rouge">group_by</code> variables</li>
</ul>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="c1"># from earlier
</span><span class="n">coverages</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read_tsv</span><span class="p">(</span><span class="s2">"../data/coverages.tsv"</span><span class="p">,</span><span class="w"> </span><span class="n">col_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"Accession"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Coverage"</span><span class="p">))</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## Parsed with column specification:
## cols(
##   Accession = col_character(),
##   Coverage = col_integer()
## )
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="c1"># use mutate to create new columns
</span><span class="n">cov</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">mutate</span><span class="p">(</span><span class="n">coverages</span><span class="p">,</span><span class="w"> </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Coverage</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">50</span><span class="p">)</span><span class="w">
</span><span class="n">cov</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## # A tibble: 216 × 3
##    Accession Coverage  high
##        &lt;chr&gt;    &lt;int&gt; &lt;lgl&gt;
## 1       Aa-0       41 FALSE
## 2      Abd-0       67  TRUE
## 3       Ag-0       47 FALSE
## 4       Ak-1       38 FALSE
## 5     Alst-1       42 FALSE
## 6    Altai-5       42 FALSE
## 7     Amel-1       60  TRUE
## 8       An-1       39 FALSE
## 9      Ang-0       48 FALSE
## 10  Anholt-1       53  TRUE
## # ... with 206 more rows
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="c1"># group rows by common variables
</span><span class="n">cov</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">group_by</span><span class="p">(</span><span class="n">cov</span><span class="p">,</span><span class="w"> </span><span class="n">high</span><span class="p">)</span><span class="w">
</span><span class="n">cov</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">mutate</span><span class="p">(</span><span class="n">cov</span><span class="p">,</span><span class="w"> </span><span class="n">group_average</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">mean</span><span class="p">(</span><span class="n">Coverage</span><span class="p">))</span><span class="w">
</span><span class="n">cov</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## Source: local data frame [216 x 4]
## Groups: high [2]
## 
##    Accession Coverage  high group_average
##        &lt;chr&gt;    &lt;int&gt; &lt;lgl&gt;         &lt;dbl&gt;
## 1       Aa-0       41 FALSE      37.32609
## 2      Abd-0       67  TRUE      70.79487
## 3       Ag-0       47 FALSE      37.32609
## 4       Ak-1       38 FALSE      37.32609
## 5     Alst-1       42 FALSE      37.32609
## 6    Altai-5       42 FALSE      37.32609
## 7     Amel-1       60  TRUE      70.79487
## 8       An-1       39 FALSE      37.32609
## 9      Ang-0       48 FALSE      37.32609
## 10  Anholt-1       53  TRUE      70.79487
## # ... with 206 more rows
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="c1"># use select to choose columns
</span><span class="n">select</span><span class="p">(</span><span class="n">cov</span><span class="p">,</span><span class="w"> </span><span class="n">Accession</span><span class="p">,</span><span class="w"> </span><span class="n">group_average</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## Adding missing grouping variables: `high`
</code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## Source: local data frame [216 x 3]
## Groups: high [2]
## 
##     high Accession group_average
##    &lt;lgl&gt;     &lt;chr&gt;         &lt;dbl&gt;
## 1  FALSE      Aa-0      37.32609
## 2   TRUE     Abd-0      70.79487
## 3  FALSE      Ag-0      37.32609
## 4  FALSE      Ak-1      37.32609
## 5  FALSE    Alst-1      37.32609
## 6  FALSE   Altai-5      37.32609
## 7   TRUE    Amel-1      70.79487
## 8  FALSE      An-1      37.32609
## 9  FALSE     Ang-0      37.32609
## 10  TRUE  Anholt-1      70.79487
## # ... with 206 more rows
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="c1"># won't let you drop a column if the data is grouped by that column
</span><span class="n">cov</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ungroup</span><span class="p">(</span><span class="n">cov</span><span class="p">)</span><span class="w">
</span><span class="n">select</span><span class="p">(</span><span class="n">cov</span><span class="p">,</span><span class="w"> </span><span class="n">Accession</span><span class="p">,</span><span class="w"> </span><span class="n">group_average</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## # A tibble: 216 × 2
##    Accession group_average
##        &lt;chr&gt;         &lt;dbl&gt;
## 1       Aa-0      37.32609
## 2      Abd-0      70.79487
## 3       Ag-0      37.32609
## 4       Ak-1      37.32609
## 5     Alst-1      37.32609
## 6    Altai-5      37.32609
## 7     Amel-1      70.79487
## 8       An-1      37.32609
## 9      Ang-0      37.32609
## 10  Anholt-1      70.79487
## # ... with 206 more rows
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="c1"># can alse choose what NOT to include
</span><span class="n">select</span><span class="p">(</span><span class="n">cov</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">high</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## # A tibble: 216 × 3
##    Accession Coverage group_average
##        &lt;chr&gt;    &lt;int&gt;         &lt;dbl&gt;
## 1       Aa-0       41      37.32609
## 2      Abd-0       67      70.79487
## 3       Ag-0       47      37.32609
## 4       Ak-1       38      37.32609
## 5     Alst-1       42      37.32609
## 6    Altai-5       42      37.32609
## 7     Amel-1       60      70.79487
## 8       An-1       39      37.32609
## 9      Ang-0       48      37.32609
## 10  Anholt-1       53      70.79487
## # ... with 206 more rows
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="c1"># sort using arrange
</span><span class="n">arrange</span><span class="p">(</span><span class="n">cov</span><span class="p">,</span><span class="w"> </span><span class="n">Coverage</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## # A tibble: 216 × 4
##        Accession Coverage  high group_average
##            &lt;chr&gt;    &lt;int&gt; &lt;lgl&gt;         &lt;dbl&gt;
## 1           Koln       11 FALSE      37.32609
## 2           Se-0       15 FALSE      37.32609
## 3           Ra-0       17 FALSE      37.32609
## 4          Wei-0       17 FALSE      37.32609
## 5          Old-1       18 FALSE      37.32609
## 6          Pu2-8       19 FALSE      37.32609
## 7         Ragl-1       19 FALSE      37.32609
## 8          Zdr-1       19 FALSE      37.32609
## 9           Bg-2       20 FALSE      37.32609
## 10 Buckhorn-Pass       20 FALSE      37.32609
## # ... with 206 more rows
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">arrange</span><span class="p">(</span><span class="n">cov</span><span class="p">,</span><span class="w"> </span><span class="n">desc</span><span class="p">(</span><span class="n">Coverage</span><span class="p">))</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## # A tibble: 216 × 4
##    Accession Coverage  high group_average
##        &lt;chr&gt;    &lt;int&gt; &lt;lgl&gt;         &lt;dbl&gt;
## 1       Br-0      133  TRUE      70.79487
## 2      Lan-0      127  TRUE      70.79487
## 3       Kb-0      110  TRUE      70.79487
## 4       El-0      107  TRUE      70.79487
## 5       La-0      102  TRUE      70.79487
## 6       Kl-5      100  TRUE      70.79487
## 7      Kar-1       97  TRUE      70.79487
## 8      Gel-1       96  TRUE      70.79487
## 9      Nok-3       96  TRUE      70.79487
## 10  Rennes-1       96  TRUE      70.79487
## # ... with 206 more rows
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="c1"># useful for coordinates
</span><span class="n">dmrs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read_tsv</span><span class="p">(</span><span class="s2">"../data/c_dmrs.tsv.gz"</span><span class="p">,</span><span class="w"> </span><span class="n">col_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"chr"</span><span class="p">,</span><span class="w"> </span><span class="s2">"start"</span><span class="p">,</span><span class="w"> </span><span class="s2">"end"</span><span class="p">))</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## Parsed with column specification:
## cols(
##   chr = col_character(),
##   start = col_integer(),
##   end = col_integer()
## )
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">dmrs</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## # A tibble: 13,484 × 3
##      chr  start    end
##    &lt;chr&gt;  &lt;int&gt;  &lt;int&gt;
## 1   chr1    792    911
## 2   chr1  22665  22807
## 3   chr1  41392  41445
## 4   chr1  71457  71476
## 5   chr1  76839  77252
## 6   chr1 193333 193525
## 7   chr1 255844 255893
## 8   chr1 291887 291998
## 9   chr1 355668 356037
## 10  chr1 428072 428073
## # ... with 13,474 more rows
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">arrange</span><span class="p">(</span><span class="n">dmrs</span><span class="p">,</span><span class="w"> </span><span class="n">chr</span><span class="p">,</span><span class="w"> </span><span class="n">desc</span><span class="p">(</span><span class="n">start</span><span class="p">))</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## # A tibble: 13,484 × 3
##      chr    start      end
##    &lt;chr&gt;    &lt;int&gt;    &lt;int&gt;
## 1   chr1 30417725 30417837
## 2   chr1 30416592 30416901
## 3   chr1 30416104 30416138
## 4   chr1 30391058 30391076
## 5   chr1 30390110 30390285
## 6   chr1 30388868 30389254
## 7   chr1 30388263 30388288
## 8   chr1 30376878 30377010
## 9   chr1 30345730 30346096
## 10  chr1 30295026 30295310
## # ... with 13,474 more rows
</code></pre>
</div>

<h3 id="joining-tables">Joining tables</h3>

<p><code class="highlighter-rouge">dplyr</code> also has join functions.</p>

<!-- ![](../images/joins.png) -->
<p><img src="/figure/./r_demo/joins.png" width="500" /></p>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">te_insertion_counts</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read_tsv</span><span class="p">(</span><span class="s2">"../data/insertion_counts.tsv"</span><span class="p">,</span><span class="w"> </span><span class="n">col_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"Accession"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Insertions"</span><span class="p">))</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## Parsed with column specification:
## cols(
##   Accession = col_character(),
##   Insertions = col_integer()
## )
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">left_join</span><span class="p">(</span><span class="n">coverages</span><span class="p">,</span><span class="w"> </span><span class="n">te_insertion_counts</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## Joining, by = "Accession"
</code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## # A tibble: 216 × 3
##    Accession Coverage Insertions
##        &lt;chr&gt;    &lt;int&gt;      &lt;int&gt;
## 1       Aa-0       41        367
## 2      Abd-0       67        420
## 3       Ag-0       47        443
## 4       Ak-1       38        305
## 5     Alst-1       42        379
## 6    Altai-5       42        417
## 7     Amel-1       60        330
## 8       An-1       39        313
## 9      Ang-0       48        350
## 10  Anholt-1       53        343
## # ... with 206 more rows
</code></pre>
</div>

<h2 id="put-it-all-together">Put it all together</h2>

<p>In unix, the pipe lets you use output from one program/function as input to another. This lets you chain together commands an eliminate intermediate files. It also provides the ability to start one process before the previous has finished. For example you can pipe the output of bowtie into samtools and convert straight to bam without having to wait for the mapping to finish.</p>

<p>The same can be done in R with any function that takes and returns a dataframe. In unix the pipe character is <code class="highlighter-rouge">|</code>, in R it is <code class="highlighter-rouge">%&gt;%</code>. The pipe is part of the <code class="highlighter-rouge">magrittr</code> package, but is also loaded any time you load <code class="highlighter-rouge">readr</code>, <code class="highlighter-rouge">dplyr</code>, or <code class="highlighter-rouge">tidyr</code>.</p>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read_tsv</span><span class="p">(</span><span class="s2">"../data/coverages.tsv"</span><span class="p">,</span><span class="w"> </span><span class="n">col_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"Accession"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Coverage"</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">mutate</span><span class="p">(</span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Coverage</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">50</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">group_by</span><span class="p">(</span><span class="n">high</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">mutate</span><span class="p">(</span><span class="n">group_av</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">mean</span><span class="p">(</span><span class="n">Coverage</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">left_join</span><span class="p">(</span><span class="n">te_insertion_counts</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## Parsed with column specification:
## cols(
##   Accession = col_character(),
##   Coverage = col_integer()
## )
</code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## Joining, by = "Accession"
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">df</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## Source: local data frame [216 x 5]
## Groups: high [?]
## 
##    Accession Coverage  high group_av Insertions
##        &lt;chr&gt;    &lt;int&gt; &lt;lgl&gt;    &lt;dbl&gt;      &lt;int&gt;
## 1       Aa-0       41 FALSE 37.32609        367
## 2      Abd-0       67  TRUE 70.79487        420
## 3       Ag-0       47 FALSE 37.32609        443
## 4       Ak-1       38 FALSE 37.32609        305
## 5     Alst-1       42 FALSE 37.32609        379
## 6    Altai-5       42 FALSE 37.32609        417
## 7     Amel-1       60  TRUE 70.79487        330
## 8       An-1       39 FALSE 37.32609        313
## 9      Ang-0       48 FALSE 37.32609        350
## 10  Anholt-1       53  TRUE 70.79487        343
## # ... with 206 more rows
</code></pre>
</div>

<h2 id="plotting">Plotting</h2>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span><span class="w">

</span><span class="n">ggplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">Coverage</span><span class="p">,</span><span class="w"> </span><span class="n">Insertions</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">geom_smooth</span><span class="p">(</span><span class="n">method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"lm"</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<p><img src="/figure/./r_demo/ggplot-1.png" alt="plot of chunk ggplot" /></p>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">ggplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">Coverage</span><span class="p">,</span><span class="w"> </span><span class="n">Insertions</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">high</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">geom_smooth</span><span class="p">(</span><span class="n">method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"lm"</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<p><img src="/figure/./r_demo/ggplot-2.png" alt="plot of chunk ggplot" /></p>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">ggplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">Coverage</span><span class="p">,</span><span class="w"> </span><span class="n">Insertions</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">facet_wrap</span><span class="p">(</span><span class="o">~</span><span class="n">high</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme_bw</span><span class="p">()</span><span class="w">
</span></code></pre>
</div>

<p><img src="/figure/./r_demo/ggplot-3.png" alt="plot of chunk ggplot" /></p>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">ggplot</span><span class="p">(</span><span class="n">tidy_data</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">experiment</span><span class="p">,</span><span class="w"> </span><span class="n">mC</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_jitter</span><span class="p">(</span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">theme_bw</span><span class="p">()</span><span class="w">
</span></code></pre>
</div>

<p><img src="/figure/./r_demo/ggplot-4.png" alt="plot of chunk ggplot" /></p>

<p>Easily create interactive graphics</p>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">plotly</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## 
## Attaching package: 'plotly'
</code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## The following object is masked from 'package:ggplot2':
## 
##     last_plot
</code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## The following object is masked from 'package:graphics':
## 
##     layout
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ggplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">Coverage</span><span class="p">,</span><span class="w"> </span><span class="n">Insertions</span><span class="p">,</span><span class="w"> </span><span class="n">label</span><span class="o">=</span><span class="n">Accession</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme_bw</span><span class="p">()</span><span class="w">

</span><span class="n">ggplotly</span><span class="p">(</span><span class="n">p</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## Error in loadNamespace(name): there is no package called 'webshot'
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="c1"># from earlier
</span><span class="n">dmrs</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## # A tibble: 13,484 × 3
##      chr  start    end
##    &lt;chr&gt;  &lt;int&gt;  &lt;int&gt;
## 1   chr1    792    911
## 2   chr1  22665  22807
## 3   chr1  41392  41445
## 4   chr1  71457  71476
## 5   chr1  76839  77252
## 6   chr1 193333 193525
## 7   chr1 255844 255893
## 8   chr1 291887 291998
## 9   chr1 355668 356037
## 10  chr1 428072 428073
## # ... with 13,474 more rows
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">ggplot</span><span class="p">(</span><span class="n">dmrs</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">start</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_density</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">facet_wrap</span><span class="p">(</span><span class="o">~</span><span class="n">chr</span><span class="p">)</span><span class="w">
</span></code></pre>
</div>

<p><img src="/figure/./r_demo/more_plots-1.png" alt="plot of chunk more_plots" /></p>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="c1"># cleaned up
</span><span class="n">ggplot</span><span class="p">(</span><span class="n">dmrs</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">start</span><span class="o">/</span><span class="m">1000</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_density</span><span class="p">(</span><span class="n">adjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="o">/</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"blue"</span><span class="p">,</span><span class="w"> </span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">facet_wrap</span><span class="p">(</span><span class="o">~</span><span class="n">chr</span><span class="p">,</span><span class="w"> </span><span class="n">scales</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"free"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">theme_classic</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">xlab</span><span class="p">(</span><span class="s2">"Position (kb)"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">ggtitle</span><span class="p">(</span><span class="s2">"C-DMR Positions"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">axis.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">45</span><span class="p">,</span><span class="w"> </span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">))</span><span class="w">
</span></code></pre>
</div>

<p><img src="/figure/./r_demo/more_plots-2.png" alt="plot of chunk more_plots" /></p>

<h2 id="reporting">Reporting</h2>

<p>Use <a href="http://rmarkdown.rstudio.com/">RMarkdown</a> (included in RStudio). Press Knit HTML / Knit PDF to render the output document.</p>

<h3 id="displaying-nice-tables">Displaying nice tables</h3>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="c1"># from before
</span><span class="n">data</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## # A tibble: 500 × 5
##    position experiment_1 experiment_2 experiment_3 experiment_4
##       &lt;int&gt;        &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;
## 1         1    0.2947252    0.0000000   0.69056274    0.2240412
## 2         2    0.2624428    0.1944636   0.02995589    0.2823043
## 3         3    0.2227268    0.1696682   0.64657079    0.3024177
## 4         4    0.1863447    0.3532347   0.67337368    0.2075265
## 5         5    0.2532761    0.7560800   0.52507202    0.2844748
## 6         6    0.1269957    0.0000000   0.69156885    0.3399855
## 7         7    0.1476328    0.3013359   0.19931927    0.4005465
## 8         8    0.2202941    0.1325935   0.31909674    0.3487259
## 9         9    0.1886488    0.4738528   0.69180550    0.4154950
## 10       10    0.1971425    0.4245025   0.99153173    0.3800341
## # ... with 490 more rows
</code></pre>
</div>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="c1"># using kable
</span><span class="n">head</span><span class="p">(</span><span class="n">data</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">knitr</span><span class="o">::</span><span class="n">kable</span><span class="p">()</span><span class="w">
</span></code></pre>
</div>

<table>
  <thead>
    <tr>
      <th style="text-align: right">position</th>
      <th style="text-align: right">experiment_1</th>
      <th style="text-align: right">experiment_2</th>
      <th style="text-align: right">experiment_3</th>
      <th style="text-align: right">experiment_4</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: right">1</td>
      <td style="text-align: right">0.2947252</td>
      <td style="text-align: right">0.0000000</td>
      <td style="text-align: right">0.6905627</td>
      <td style="text-align: right">0.2240412</td>
    </tr>
    <tr>
      <td style="text-align: right">2</td>
      <td style="text-align: right">0.2624428</td>
      <td style="text-align: right">0.1944636</td>
      <td style="text-align: right">0.0299559</td>
      <td style="text-align: right">0.2823043</td>
    </tr>
    <tr>
      <td style="text-align: right">3</td>
      <td style="text-align: right">0.2227268</td>
      <td style="text-align: right">0.1696682</td>
      <td style="text-align: right">0.6465708</td>
      <td style="text-align: right">0.3024177</td>
    </tr>
    <tr>
      <td style="text-align: right">4</td>
      <td style="text-align: right">0.1863447</td>
      <td style="text-align: right">0.3532347</td>
      <td style="text-align: right">0.6733737</td>
      <td style="text-align: right">0.2075265</td>
    </tr>
    <tr>
      <td style="text-align: right">5</td>
      <td style="text-align: right">0.2532761</td>
      <td style="text-align: right">0.7560800</td>
      <td style="text-align: right">0.5250720</td>
      <td style="text-align: right">0.2844748</td>
    </tr>
    <tr>
      <td style="text-align: right">6</td>
      <td style="text-align: right">0.1269957</td>
      <td style="text-align: right">0.0000000</td>
      <td style="text-align: right">0.6915689</td>
      <td style="text-align: right">0.3399855</td>
    </tr>
  </tbody>
</table>

<h3 id="code-block-options">Code block options</h3>

<p>Can choose to show everything, show only output (eg plots), show nothing, show only warnings, etc..</p>

<h3 id="output-options">Output options</h3>

<p>Output can be rendered as html, pdf, or word. You can also choose a theme, and whether to show a table of contents. One important feature is the ability to have self-contained html documents by including  <code class="highlighter-rouge">self_contained: true</code> in the rmarkdown header. This will let you send the html file to others and still have all the plots appear correct. This should be set as the default for html output.</p>

<h3 id="running-code-in-other-languages">Running code in other languages</h3>

<p>It is possible to have code chunks that run in languages other than R. If you look at the first code chunk in this document, you will see that it runs a shell command (git). To find out more, read about <a href="http://yihui.name/knitr/demo/engines/">language engines in knitr</a>.</p>

<p>Here is an example running python:</p>

<div class="language-python highlighter-rouge"><pre class="highlight"><code><span class="n">chromosomes</span> <span class="o">=</span> <span class="p">[</span><span class="s">"chr"</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">)</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">xrange</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">6</span><span class="p">)]</span>
<span class="k">print</span><span class="p">(</span><span class="n">chromosomes</span><span class="p">)</span>
</code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## ['chr1', 'chr2', 'chr3', 'chr4', 'chr5']
</code></pre>
</div>

<p>To use objects created by running code in other languages (for example, if we wanted to use the <code class="highlighter-rouge">chromosomes</code> variable created in the above python chunk), you will usually need to write the results to a file and then read into R from the file. There are complicated environment variable things you can do to <a href="https://raw.githubusercontent.com/yihui/knitr-examples/master/106-polyglot.Rmd">share objects</a> between languages, but these may not be worth the extra effort.</p>

<h2 id="further-reading">Further reading</h2>

<p><a href="http://r4ds.had.co.nz/">http://r4ds.had.co.nz/</a></p>

<h2 id="session-info">Session info</h2>

<div class="language-r highlighter-rouge"><pre class="highlight"><code><span class="n">devtools</span><span class="o">::</span><span class="n">session_info</span><span class="p">()</span><span class="w">
</span></code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## Session info --------------------------------------------------------------
</code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>##  setting  value                       
##  version  R version 3.3.1 (2016-06-21)
##  system   x86_64, darwin15.5.0        
##  ui       unknown                     
##  language (EN)                        
##  collate  en_AU.UTF-8                 
##  tz       Australia/Perth             
##  date     2016-09-22
</code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>## Packages ------------------------------------------------------------------
</code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>##  package      * version date       source                      
##  assertthat     0.1     2013-12-06 CRAN (R 3.3.0)              
##  base64enc      0.1-3   2015-07-28 CRAN (R 3.3.1)              
##  chron          2.3-47  2015-06-24 CRAN (R 3.3.0)              
##  colorspace     1.2-6   2015-03-11 CRAN (R 3.3.0)              
##  data.table   * 1.9.6   2015-09-19 CRAN (R 3.3.1)              
##  DBI            0.5-1   2016-09-10 CRAN (R 3.3.1)              
##  devtools       1.12.0  2016-06-24 CRAN (R 3.3.1)              
##  digest         0.6.10  2016-08-02 CRAN (R 3.3.1)              
##  dplyr        * 0.5.0   2016-06-24 CRAN (R 3.3.1)              
##  evaluate       0.9     2016-04-29 CRAN (R 3.3.0)              
##  formatR        1.4     2016-05-09 CRAN (R 3.3.1)              
##  ggplot2      * 2.1.0   2016-03-01 CRAN (R 3.3.1)              
##  gridExtra      2.2.1   2016-02-29 CRAN (R 3.3.1)              
##  gtable         0.2.0   2016-02-26 CRAN (R 3.3.0)              
##  highr          0.6     2016-05-09 CRAN (R 3.3.0)              
##  htmltools      0.3.5   2016-03-21 CRAN (R 3.3.0)              
##  htmlwidgets    0.7     2016-08-02 CRAN (R 3.3.1)              
##  httr           1.2.1   2016-07-03 CRAN (R 3.3.1)              
##  jsonlite       1.1     2016-09-14 CRAN (R 3.3.1)              
##  knitr          1.14.4  2016-09-22 Github (yihui/knitr@54ea64f)
##  labeling       0.3     2014-08-23 CRAN (R 3.3.0)              
##  lazyeval       0.2.0   2016-06-12 CRAN (R 3.3.0)              
##  magrittr       1.5     2014-11-22 CRAN (R 3.3.0)              
##  memoise        1.0.0   2016-01-29 CRAN (R 3.3.0)              
##  munsell        0.4.3   2016-02-13 CRAN (R 3.3.0)              
##  plotly       * 3.6.0   2016-05-18 CRAN (R 3.3.1)              
##  plyr           1.8.4   2016-06-08 CRAN (R 3.3.0)              
##  R6             2.1.3   2016-08-19 CRAN (R 3.3.1)              
##  RColorBrewer * 1.1-2   2014-12-07 CRAN (R 3.3.0)              
##  Rcpp           0.12.7  2016-09-05 CRAN (R 3.3.1)              
##  readr        * 1.0.0   2016-08-03 CRAN (R 3.3.1)              
##  scales         0.4.0   2016-02-26 CRAN (R 3.3.0)              
##  stringi        1.1.1   2016-05-27 CRAN (R 3.3.0)              
##  stringr        1.1.0   2016-08-19 CRAN (R 3.3.1)              
##  tibble         1.2     2016-08-26 CRAN (R 3.3.1)              
##  tidyr        * 0.6.0   2016-08-12 CRAN (R 3.3.1)              
##  viridis        0.3.4   2016-03-12 CRAN (R 3.3.1)              
##  withr          1.0.2   2016-06-20 CRAN (R 3.3.1)              
##  yaml           2.1.13  2014-06-12 CRAN (R 3.3.0)
</code></pre>
</div>


</div>


<div id="disqus_thread"></div>
<script>

  /**
  *  RECOMMENDED CONFIGURATION VARIABLES: EDIT AND UNCOMMENT THE SECTION BELOW TO INSERT DYNAMIC VALUES FROM YOUR PLATFORM OR CMS.
  *  LEARN WHY DEFINING THESE VARIABLES IS IMPORTANT: https://disqus.com/admin/universalcode/#configuration-variables*/
  /*
  var disqus_config = function () {
  this.page.url = PAGE_URL;  // Replace PAGE_URL with your page's canonical URL variable
  this.page.identifier = PAGE_IDENTIFIER; // Replace PAGE_IDENTIFIER with your page's unique identifier variable
  };
  */
  (function() { // DON'T EDIT BELOW THIS LINE
  var d = document, s = d.createElement('script');
  s.src = '//timoast-github-io.disqus.com/embed.js';
  s.setAttribute('data-timestamp', +new Date());
  (d.head || d.body).appendChild(s);
  })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>



      <!-- <div class="footer">
        <p>
          &copy; 2017. All rights reserved.
        </p>
      </div> -->
    </div>

  </body>
</html>
