<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">

  <head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  <!-- Enable responsiveness on mobile devices-->
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>
    
      Tim Stuart &middot; 
    
  </title>

  <!-- CSS -->
  <link rel="stylesheet" href="/public/css/poole.css">
  <link rel="stylesheet" href="/public/css/syntax.css">

  <!-- Icons -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/public/apple-touch-icon-precomposed.png">
  <link rel="shortcut icon" href="/public/favicon.ico">

  <!-- RSS -->
  <link rel="alternate" type="application/rss+xml" title="RSS" href="/atom.xml">
</head>

  <script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-54133406-1', 'auto');
  ga('send', 'pageview');

</script>


  <body>

    <div class="container content">
      <div class="masthead">
        <h3 class="masthead-title">
          <a href="/" title="Home">Tim Stuart</a>
          <small></small>

          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/about">About</a></small>
          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/archive">Archive</a></small>
          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/code">Code</a></small>
          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/publications">Publications</a></small>
          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/atom.xml">Feed</a></small>
          
        </h3>
      </div>

      <div class="posts">
  
  <div class="post">
    <h1 class="post-title">
      <a href="/2016/05/24/smRNA/">
        TE annotation using smRNAs
      </a>
    </h1>

    <span class="post-date">24 May 2016</span>

    <p>I want to annotate TEs in the Landsberg <em>erecta de novo</em> assembly genome (made with PacBio data). This is to address some reviewer comments for a paper I’m currently <a href="http://biorxiv.org/content/early/2016/02/21/039511">working on</a>.</p>

<p>There are several possible ways to do this, and one is using smRNA reads that mark TE positions in the genome. This approach was described in a recent paper, which also includes code to perform the TE annotation given smRNA data (1).</p>

<p>According to the paper, this works best using smRNA libraries pooled from lots of different tissues, as the diversity in the library is most important, and some TEs accumulate smRNAs only in certain tissues (eg. the vegetative cell nucleus of the pollen, endosperm, or columella cells).</p>

<p>Looking for as L<em>er</em> smRNA data, I found two different experiments that I will pool together and use. These are 2 week old whole seedlings (2) and whole seed smRNAs 6 days after pollination (3).</p>

<p>To download data:</p>

<p>```
wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByExp/sra/SRX%2FSRX029%2FSRX029679/SRR070690/SRR070690.sra # Groszmann, ref 2
wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByExp/sra/SRX%2FSRX384%2FSRX384740/SRR1039937/SRR1039937.sra # Pignatta, ref 3</p>

<p>fastq-dump *.sra
rm *.sra
pigz *.fastq
```</p>

<h2 id="qc-and-adapter-trimming">QC and adapter trimming</h2>

<p>Much of what I’m doing in this step comes from <a href="http://www.ark-genomics.org/events-online-training-eu-training-course/adapter-and-quality-trimming-illumina-data">this guide</a>.</p>

<p>First we’ll run FastQC:</p>

<p><code>
fastqc *.fastq.gz
</code></p>

<p>Looking at the overrepresented sequences section of the report, we can find the adapter sequence:</p>

<p>For the Groszmann (2) data:</p>

<p><code>
TCGTATGCCGTCTTCTGCTTGAAAAAAAAAAAAAAA    13924   0.2724698598318295  Illumina Single End Adapter 1 (100% over 21bp)
</code>
The poly A sequence at the end is not part of the adapter (as far as I know), but it shouldn’t be in our read either. We can take a look at some reads with adapters:</p>

<p>We can also look for the adapter sequences in some of the data:</p>

<p><code>
zcat SRR070690.fastq.gz | head -n 40000 | grep --color=always TCGTATGCCGTCTTCTGCTTG
</code></p>

<p><img src="/Users/timstuart/Documents/GitHub/timoast.github.io/assets/smRNA_terminal.png" alt="grep_output" /></p>

<p>This shows that many of the reads are adapter dimers, since lots of reads start with adapter, which is expected in a smRNA library. The full sequence of this adapter can be found in the FastQC folder under FastQC/Configuration/contaminant_list.txt.</p>

<p>The Pignatta (3) data has lots of different sequences identified:</p>

<p><code>
GTTCTACAGTCCGACGATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAA  3163237 7.821875679772812   Illumina PCR Primer Index 1 (96% over 28bp)
TTCTACAGTCCGACGATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAA  2062049 5.0989195319856995  RNA PCR Primer, Index 47 (96% over 29bp)
AGTTCTACAGTCCGACGATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAA  2028870 5.016876354955594   Illumina PCR Primer Index 1 (96% over 28bp)
...
</code></p>

<p>This may be because the reads are longer (50 bp). Mostly it’s Illumina PCR Primer Index 1, RNA PCR Primer Index 47, Illumina Single End Adapter 1, RNA PCR Primer Index 30, and Illumina Single End Adapter 2.</p>

<p>In the paper, they say they trimmed using “TCGTATGCCGTCTTCTGCTTG”, but I tried that and found there was still a lot of adapter content. They were using <code>fastx_clipper</code> which is a bit more aggressive, which could explain the difference.</p>

<p>Running cutadapt will trim the adapter sequences. I had to play around for a while before I was able to remove most of the adapter sequences from the Pignatta data with some aggressive trimming. It’s not pretty but it gets the job done.</p>

<p>```
# Groszmann data
cutadapt -a TCGTATGCCGTCTTCTGCTTG -o SRR070690.trim.fq.gz SRR070690.fastq.gz</p>

<h1 id="pignatta-data">Pignatta data</h1>
<p>cutadapt -a GTTCTACAGTCCGACGATCTCGTATGCCGTCTTCTGCTTG \
         -a TCTACAGTCCGACGATCTCGTATGCCGTCTTCTGCTTG \
         -a GTTCTACAGTCCGACGGATCTCGTATGCCGTCTTCTGCTTG \
         -a AGTTCTACAGTCCGACGGATCTCGTATGCCGTCTTCTGCTTG \
         -a ATTAACGCGGCCGCTCTACAATAGTGATCGTATGCCGTCTTCTGCTTG \
         -a GTGTAGGGATCCAAATCGTATGCCGTCTTCTGCTTG \
         -a CAGTCCGACGATCTCGTATGCCGTCTTCTGCTTG\
         -a ACAGTCCGACGATCTCGTATGCCGTCTTCTGCTTG \
         -a GCCGCTCTACAATAGTGATCGTATGCCGTCTTCTGCTTG \
         -a TCGTATGCCGTCTTCTGCTTG \
         -o SRR1039937.trim.fq.gz SRR1039937.fastq.gz 
```</p>

<p>Running sickle will trim low quality bases. The default minimum length read kept by sickle is 20 bp. I’m using the <code>-l</code> flag to set this to 10, as smRNA data will naturally have shorter reads in it, although I’m not sure how useful it really is to keep anything less than 20. For what I’m using this data for, only 24 nt smRNAs are used.</p>

<p><code>
sickle se -f SRR070690.trim.fq.gz -t sanger -o SRR070690_qc.fq.gz -l 10 -g
sickle se -f SRR1039937.trim.fq.gz -t sanger -o SRR1039937_qc.fq.gz -l 10 -g
</code></p>

<p>If we then look at the fastqc output for the trimmed reads, we can see peaks in read lengths at 21 and 24 nt, corresponding to the most abundant smRNA size classes in plants.</p>

<p><img src="/Users/timstuart/Documents/GitHub/timoast.github.io/assets/Groszmann_fastqc.png" alt="Groszmann_fastqc" /> <img src="/Users/timstuart/Documents/GitHub/timoast.github.io/assets/Pignatta_fastqc.png" alt="pignatta_fastqc" /></p>

<h2 id="run-tasr">Run TASR</h2>

<p>TASR is the analysis pipeline for smRNA-based annotation of TEs, descibed in (1). It can be downloaded from <a href="https://sourceforge.net/projects/tasr-pipeline/">https://sourceforge.net/projects/tasr-pipeline/</a></p>

<p>The TASR pipeline wants smRNA data in fasta rather than fastq format. To convert, we do:</p>

<p><code>
seqtk seq -A SRR070690_qc.fq.gz &gt; SRR070690_qc.fa
seqtk seq -A SRR1039937_qc.fq.gz &gt; SRR1039937_qc.fa
cat SRR1039937_qc.fa SRR070690_qc.fa &gt; reads.fa
</code></p>

<p>It also wants just the unique smRNAs, so we need to collapse our fasta file:</p>

<p><code>
fastx_collapser -i reads.fa -o collapsed_reads.fa
</code></p>

<p>First make sure the dependencies are installed. I had to install <a href="http://www.drive5.com/usearch/">usearch</a>, <a href="https://tandem.bu.edu/trf/trf.download.html">trf</a> (the legacy version for glibc &lt;= 2.12), and <a href="ftp://pbil.univ-lyon1.fr/pub/logiciel/silix/">silix</a>. Make sure you have both silix and silix-split in your path. Also, you need to make sure your version of sort has the parallel option, or remove the –parallel flag from all the sort calls in the perl script like this (there are more):</p>

<p><code>
SAMTOBED("tmp$time/24.sirna.sam");
system(
    "sort --parallel$CPU -k1,1 -k2,2n tmp$time/24.sirna.sam.bed &gt;tmp$time/24.sirna.sam.sorted.bed"
);
</code></p>

<p>To</p>

<p><code>
SAMTOBED("tmp$time/24.sirna.sam");
system(
    "sort -k1,1 -k2,2n tmp$time/24.sirna.sam.bed &gt;tmp$time/24.sirna.sam.sorted.bed"
);
</code></p>

<p>Now run:</p>

<p><code>
TASR.v.1.1.pl -ref ler_pacbio.fa -sfile collapsed_reads.fa -usearchv /home/tstuart/working_data/Tools/usearch -cpu 10
</code></p>

<p>I’m just using the default values for all parameters to start with.</p>

<p>This gives the following output:</p>

<h2 id="references">References</h2>

<ol>
  <li>
    <p>Baidouri El M, Kim KD, Abernathy B, Arikit S, Maumus F, Panaud O, et al. A new approach for annotation of transposable elements using small RNA mapping. Nucleic Acids Research. 2015;43: e84–e84.</p>
  </li>
  <li>
    <p>Groszmann M, Greaves IK, Albertyn ZI, Scofield GN, Peacock WJ, Dennis ES. Changes in 24-nt siRNA levels in Arabidopsis hybrids suggest an epigenetic contribution to hybrid vigor. Proceedings of the National Academy of Sciences. National Acad Sciences; 2011;108: 2617–2622. doi:10.1073/pnas.1019217108</p>
  </li>
  <li>
    <p>Pignatta D, Erdmann RM, Scheer E, Picard CL, Bell GW, Gehring M. Natural epigenetic polymorphisms lead to intraspecific variation in Arabidopsis gene imprinting. eLife. 2014;3: e03198. doi:10.7554/eLife.03198</p>
  </li>
</ol>

  </div>
  
</div>

<div class="pagination">
  
    <a class="pagination-item older" href="/page2">Older</a>
  
  
    <span class="pagination-item newer">Newer</span>
  
</div>


      <!-- <div class="footer">
        <p>
          &copy; 2016. All rights reserved.
        </p>
      </div> -->
    </div>

  </body>
</html>
