---
layout: post
title: Highly Multiplexed Subcellular RNA Sequencing in Situ
date: 2014-03-04 16:18:11.000000000 +08:00
categories: []
tags:
- NGS
- RNA
- sequencing
status: publish
type: post
published: true
meta:
  _publicize_pending: '1'
author:
---
<p>An improved method capable of sequencing RNA inside fixed cells (FISSEQ) was published in the early issue of <i>Science</i> last week, titled <i>Highly Multiplexed Subcellular RNA Sequencing in Situ </i>(1). It's an interesting method, and I'm going to go over it briefly here.</p>
<p>Firstly, the method is not new; it was developed in 2003, and this is stated in the first paragraph of the paper. The original 2003 method was conceptually the same, involving sequencing of DNA bases through the incorporation of fluorescent nucleotides. However, this was only capable of sequencing libraries fixed in gel on a glass slide, and produced a measly 8 bp of sequence (2). Needless to say, huge improvements in sequencing technologies have been made over the past ten years.</p>
<p>The new method presented by Lee et al. works in the following way. Firstly, cells are fixed with formaldehyde and permeabilised, and an <em>in situ</em> reverse transcription reaction performed with random hexamers and aminoallyl dUTP. cDNA is then circularised and amplified by rolling circle amplification. The authors report that the use of RNaseA improves the circularisation of cDNAs for rolling circle amplification (an RNaseH digest is also done after the RT to degrade RNA templates). dUTPs incorporated during the RT then allow crosslinking of the amplicons, through the addition of BS(PEG)9. This improves spacial stabilisation of the library <em>in situ</em><em>. </em>Addition of BS(PEG)9 is also helpful in reducing non-specific probe binding. Next, a formamide wash removes unlinked cDNA amplicons, improving amplicon density, and fluorescent probes are hybridised to the adapter sequence. Finally, SOLiD sequencing-by-ligation can be performed, and the authors report read lengths of 27 bp (up from 8 bp in 2003), with a median error rate of 0.64% (lower than the 1.33% of <a href="http://res.illumina.com/documents/products/datasheets/datasheet_genomic_sequence.pdf" target="_blank">Illumina</a> sequencing). Additionally, this method provides subcellular localisation data for reads. The end result is a bunch of cool flashing dots inside a cell:</p>
<blockquote><p><img src="/assets/s5_reduced1.gif" width="250" height="250" /></p>
<p>Video S5 from Lee et al. (2014) (1).</p></blockquote>
<p>The authors have compared the FISSEQ method to RNA-seq and gene expression arrays, and find a reasonably good correlation between some genes (those that were moderately expressed), though overall the correlation is poor:</p>
<blockquote><p><img src="/assets/s11c.png" width="300" height="220" /><br />
Figure S11C from Lee et al. (2014) (1).</p></blockquote>
<p>There also seems to be a partial depletion of reads involved in RNA and protein processing, and the authors suggest that this could be due to some cellular structures, such as the nucleolus, being inaccessible by the FISSEQ method. I would agree that this is the case, since there were far fewer rRNA reads detected than would be expected (only 42.7% in primary fibroblasts) and it's pretty clear from some of the figures and videos that there are amplicon-free regions:</p>
<blockquote><p><img src="/assets/s6b.png" width="300" height="296" /></p>
<p>Figure S6B from Lee et al. (2014) (1).</p></blockquote>
<p>Actually, it could be a good thing that some parts of the cell are not covered by FISSEQ, particularly the nucleolus, as this seems to provide a partial rRNA depletion.</p>
<p>Overall, I think the method is an exciting one, and an impressive achievement with some promising applications in the future—the authors suggest <em>in situ </em>cell type identification based on expression profiling, and high-throughput <em>in situ </em>mutation detection. However, these applications are likely to be held back until longer read lengths can be achieved, and the protocol can simplified to reduce hands-on time. From the supplementary methods, I estimated just the SOLiD sequencing to take around two hours per cycle. Considering 27 cycles were done in these experiments, that would total 54 hours of hands-on lab time, for only 27 bp of sequence. Little mention is made of the amount of time needed for downstream informatics.</p>
<p><a href="http://www.sciencemag.org/content/early/2014/02/26/science.1250212" target="_blank">1. <span style="line-height:1.5em;">Lee, J. H., Daugharthy, E. R., Scheiman, J., Kalhor, R., Yang, J. L., Ferrante, T. C., et al. (2014). Highly Multiplexed Subcellular RNA Sequencing in Situ. Science. doi:10.1126/science.1250212</span></a><a href="http://www.sciencemag.org/content/early/2014/02/26/science.1250212" target="_blank"><br />
</a></p>
<p><a href="http://arep.med.harvard.edu/pdf/Mitra03c.pdf" target="_blank">2. Mitra, R. D., Shendure, J., Olejnik, J., Edyta-Krzymanska-Olejnik, &amp; Church, G. M. (2003). Fluorescent in situ sequencing on polymerase colonies. Analytical Biochemistry, 320(1), 55–65. doi:10.1016/S0003-2697(03)00291-4</a></p>
