<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">

  <head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  <!-- Enable responsiveness on mobile devices-->
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>
    
      A paper a day &middot; Tim Stuart
    
  </title>

  <!-- CSS -->
  <link rel="stylesheet" href="/public/css/poole.css">
  <link rel="stylesheet" href="/public/css/syntax.css">

  <!-- Icons -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/public/apple-touch-icon-precomposed.png">
  <link rel="shortcut icon" href="/public/favicon.ico">

  <!-- RSS -->
  <link rel="alternate" type="application/rss+xml" title="RSS" href="/atom.xml">
</head>

  <script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-54133406-1', 'auto');
  ga('send', 'pageview');

</script>


  <body>

    <div class="container content">
      <div class="masthead">
        <h3 class="masthead-title">
          <a href="/" title="Home">Tim Stuart</a>
          <small></small>

          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/about">About</a></small>
          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/archive">Archive</a></small>
          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/code">Code</a></small>
          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/publications">Publications</a></small>
          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/atom.xml">Feed</a></small>
          
        </h3>
      </div>

      <div class="post">
  <h1 class="post-title">A paper a day</h1>
  <span class="post-date">23 Feb 2016</span>
  <p>In an effort to read more papers this year, I’m going to read a paper (or something paper-like) each day for the remainder of the year, and post each paper below as I go.</p>

<h3 id="jun-6">Jun 6</h3>

<p>Chin C-S, Peluso P, Sedlazeck FJ, Nattestad M, Concepcion GT, Clum A, et al. Phased Diploid Genome Assembly with Single Molecule Real-Time Sequencing. bioRxiv. 2016 Jun p. 056887. <a href="http:/dx.doi.org/10.1101/056887">doi:10.1101/056887</a></p>

<h3 id="jun-5">Jun 5</h3>

<p>Quadrana L, Bortolini Silveira A, Mayhew GF, LeBlanc C, Martienssen RA, Jeddeloh JA, et al. The Arabidopsis thaliana mobilome and its impact at the species level. eLife. 2016;5: e15716. <a href="http://dx.doi.org/10.7554/eLife.15716">doi:10.7554/eLife.15716</a></p>

<h3 id="jun-4">Jun 4</h3>

<p>Taudt A, Colomé-Tatché M, Johannes F. Genetic sources of population epigenomic variation. Nature Reviews Genetics. 2016;: 1–14. <a href="http://dx.doi.org/10.1038/nrg.2016.45">doi:10.1038/nrg.2016.45</a></p>

<h3 id="jun-3">Jun 3</h3>

<p>Boeke JD, Church G, Hessel A, Kelley NJ, Arkin A, Cai Y, et al. The Genome Project–Write. Science. 2016. <a href="http://dx.doi.org/10.1126/science.aaf6850">doi:10.1126/science.aaf6850</a></p>

<h3 id="jun-2">Jun 2</h3>

<p>Pedersen BS, Layer RM, Quinlan AR. Vcfanno: fast, flexible annotation of genetic variants. Genome Biology. 2016;17: 1–9. <a href="http://dx.doi.org/10.1186/s13059-016-0973-5">doi:10.1186/s13059-016-0973-5</a></p>

<h3 id="jun-1">Jun 1</h3>

<p>Wibowo A, Becker C, Marconi G, Durr J, Price J, Hagmann J, et al. Hyperosmotic stress memory in Arabidopsis is mediated by distinct epigenetically labile sites in the genome and is restricted in the male germline by DNA glycosylase activity. eLife. 2016;5: e13546. <a href="http://dx.doi.org/10.7554/eLife.13546">doi:10.7554/eLife.13546</a></p>

<h3 id="may-31">May 31</h3>

<p>Evrony GD, Lee E, Park PJ, Walsh CA. Resolving rates of mutation in the brain using single-neuron genomics. Eddy SR, editor. eLife. 2016;5: e12966. <a href="http://dx.doi.org/10.7554/eLife.12966">doi:10.7554/eLife.12966</a></p>

<h3 id="may-30">May 30</h3>

<p>Hirsch CD, Springer NM. Transposable element influences on gene expression in plants. Biochim Biophys Acta. 2016. <a href="http://dx.doi.org/10.1016/j.bbagrm.2016.05.010">doi:10.1016/j.bbagrm.2016.05.010</a></p>

<h3 id="may-29">May 29</h3>

<p>Li C, Gu L, Gao L, Chen C, Wei C-Q, Qiu Q, et al. Concerted genomic targeting of H3K27 demethylase REF6 and chromatin-remodeling ATPase BRM in Arabidopsis. Nature Genetics; 2016;48: 687–693. <a href="http://dx.doi.org/10.1038/ng.3555">doi:10.1038/ng.3555</a></p>

<h3 id="may-28">May 28</h3>

<p>Cui X, Lu F, Qiu Q, Zhou B, Gu L, Zhang S, et al. REF6 recognizes a specific DNA sequence to demethylate H3K27me3 and regulate organ boundary formation in Arabidopsis. Nature Genetics; 2016;48: 694–699. <a href="http://dx.doi.org/10.1038/ng.3556">doi:10.1038/ng.3556</a></p>

<h3 id="may-27">May 27</h3>

<p>Liu C, Wang C, Wang G, Becker C, Zaidem M, Weigel D. Genome-wide analysis of chromatin packing in Arabidopsis thaliana at single-gene resolution. Genome Research. 2016. <a href="http://dx.doi.org/10.1101/gr.204032.116">doi:10.1101/gr.204032.116</a></p>

<h3 id="may-26">May 26</h3>

<p>Kim D-H, Sung S. The Plant Homeo Domain finger protein, VIN3-LIKE 2, is necessary for photoperiod-mediated epigenetic regulation of the floral repressor, MAF5. Proceedings of the National Academy of Sciences. 2010;107: 17029–17034. <a href="http://dx.doi.org/10.1073/pnas.1010834107">doi:10.1073/pnas.1010834107</a></p>

<h3 id="may-25">May 25</h3>

<p>Greb T, Mylne JS, Crevillén P, Geraldo N, An H, Gendall AR, et al. The PHD finger protein VRN5 functions in the epigenetic silencing of Arabidopsis FLC. Current Biology. 2007;17: 73–78. <a href="http://dx.doi.org/10.1016/j.cub.2006.11.052">doi:10.1016/j.cub.2006.11.052</a></p>

<h3 id="may-24">May 24</h3>

<p>Wang J, Yu Y, Tao F, Zhang J, Copetti D, Kudrna D, et al. DNA methylation changes facilitated evolution of genes derived from Mutator-like transposable elements. Genome Biology. 2016;: 1–20. <a href="http://dx.doi.org/10.1186/s13059-016-0954-8">doi:10.1186/s13059-016-0954-8</a></p>

<h3 id="may-23">May 23</h3>

<p>Ahmed I, Sarazin A, Bowler C, Colot V, Quesneville H. Genome-wide evidence for local DNA methylation spreading from small RNA-targeted sequences in Arabidopsis. Nucleic Acids Research. 2011;39: 6919–6931. <a href="http://dx.doi.org/10.1093/nar/gkr324">doi:10.1093/nar/gkr324</a></p>

<h3 id="may-22">May 22</h3>

<p>Shen H, He H, Li J, Chen W, Wang X, Guo L, et al. Genome-Wide Analysis of DNA Methylation and Gene Expression Changes in Two Arabidopsis Ecotypes and Their Reciprocal Hybrids. The Plant Cell. 2012;24: 875–892. <a href="http://dx.doi.org/10.1105/tpc.111.094870">doi:10.1105/tpc.111.094870</a></p>

<h3 id="may-21">May 21</h3>

<p>Wang X, Weigel D, Smith LM. Transposon variants and their effects on gene expression in Arabidopsis. PLoS Genetics. 2013;9: e1003255. <a href="http://dx.doi.org/10.1371/journal.pgen.1003255">doi:10.1371/journal.pgen.1003255</a></p>

<h3 id="may-20">May 20</h3>

<p>O’Malley RC, Huang S-SC, Song L, Lewsey MG, Bartlett A, Nery JR, et al. Cistrome and Epicistrome Features Shape the Regulatory DNA Landscape. Cell. 2016;165: 1280–1292. <a href="http://dx.doi.org/10.1016/j.cell.2016.04.038">doi:10.1016/j.cell.2016.04.038</a></p>

<h3 id="may-19">May 19</h3>

<p>Moreno-Romero J, Jiang H, Santos-González J, Köhler C. Parental epigenetic asymmetry of PRC2-mediated histone modifications in the Arabidopsis endosperm. EMBO. 2016;: 1–14. <a href="http://dx.doi.org/10.15252/embj.201593534">doi:10.15252/embj.201593534</a></p>

<h3 id="may-18">May 18</h3>

<p>Zuk O, Schaffner SF, Samocha K, Do R, Hechter E, Kathiresan S, et al. Searching for missing heritability: Designing rare variant association studies. Proc Natl Acad Sci USA. 2014; 111: E455–E464. <a href="http://dx.doi.org/10.1073/pnas.1322563111">doi:10.1073/pnas.1322563111</a></p>

<h3 id="may-17">May 17</h3>

<p>Zuk O, Hechter E, Sunyaev SR, Lander ES. The mystery of missing heritability: Genetic interactions create phantom heritability. Proc Natl Acad Sci USA. 2012;109: 1193–1198. <a href="http://dx.doi.org/10.1073/pnas.1119675109">doi:10.1073/pnas.1119675109</a></p>

<h3 id="may-16">May 16</h3>

<p>Sasaki E, Zhang P, Atwell S, Meng D, Nordborg M. “Missing” G x E Variation Controls Flowering Time in Arabidopsis thaliana. PLoS Genetics; 2015;11: e1005597. <a href="http://dx.doi.org/10.1371/journal.pgen.1005597">doi:10.1371/journal.pgen.1005597</a></p>

<h3 id="may-15">May 15</h3>

<p>Horton MW, Bodenhausen N, Beilsmith K, Meng D, Muegge BD, Subramanian S, et al. Genome-wide association study of Arabidopsis thaliana leaf microbial community. Nature Communications; 2014;5: 1–7. <a href="http://dx.doi.org/10.1038/ncomms6320">doi:10.1038/ncomms6320</a></p>

<h3 id="may-14">May 14</h3>

<p>Sudmant PH, Rausch T, Gardner EJ, Handsaker RE, Abyzov A, Huddleston J, et al. An integrated map of structural variation in 2,504 human genomes. Nature. 2015;526: 75–81. <a href="http://dx.doi.org/10.1038/nature15394">doi:10.1038/nature15394</a></p>

<h3 id="may-13">May 13</h3>

<p>Ewing AD, Kazazian HH. Whole-genome resequencing allows detection of many rare LINE-1 insertion alleles in humans. Genome Research. 2011;21: 985–990. <a href="http://dx.doi.org/10.1101/gr.114777.110">doi:10.1101/gr.114777.110</a></p>

<h3 id="may-12">May 12</h3>

<p>Ewing AD, Kazazian HH. High-throughput sequencing reveals extensive variation in human-specific L1 content in individual human genomes. Genome Research. 2010;20: 1262–1270. <a href="http://dx.doi.org/10.1101/gr.106419.110">doi:10.1101/gr.106419.110</a></p>

<h3 id="may-11">May 11</h3>

<p>Montgomery SB, Lappalainen T, Gutierrez-Arcelus M, Dermitzakis ET. Rare and Common Regulatory Variation in Population-Scale Sequenced Human Genomes. PLoS Genetics; 2011;7: e1002144–10. <a href="http://dx.doi.org/10.1371/journal.pgen.1002144">doi:10.1371/journal.pgen.1002144</a></p>

<h3 id="may-10">May 10</h3>

<p>Piegu B, Guyot R, Picault N, Roulin A, Sanyal A, Saniyal A, et al. Doubling genome size without polyploidization: dynamics of retrotransposition-driven genomic expansions in Oryza australiensis, a wild relative of rice. Genome Research. 2006;16: 1262–1269. <a href="http://dx.doi.org/10.1101/gr.5290206">doi:10.1101/gr.5290206</a></p>

<h3 id="may-9">May 9</h3>

<p>Morris AP, Zeggini E. An evaluation of statistical approaches to rare variant analysis in genetic association studies. Genet Epidemiol. 2009;34: 188–193. <a href="http://dx.doi.org/10.1002/gepi.20450">doi:10.1002/gepi.20450</a></p>

<h3 id="may-8">May 8</h3>

<p>Flutre T, Duprat E, Feuillet C, Quesneville H. Considering Transposable Element Diversification in De Novo Annotation Approaches. PLoS ONE. 2011;6: e16526. <a href="http:/dx.doi.org/10.1371/journal.pone.0016526">doi:10.1371/journal.pone.0016526</a></p>

<h3 id="may-7">May 7</h3>

<p>Hunt KA, Mistry V, Bockett NA, Ahmad T, Ban M, Barker JN, et al. Negligible impact of rare autoimmune-locus coding-region variants on missing heritability. Nature; 498: 232–235. <a href="http://dx.doi.org/10.1038/nature12170">doi:10.1038/nature12170</a></p>

<h3 id="may-6">May 6</h3>

<p>Walker AR, Lee E, Bogs J, McDavid DAJ, Thomas MR, Robinson SP. White grapes arose through the mutation of two similar and adjacent regulatory genes. The Plant Journal. 2007;49: 772–785. <a href="http://dx.doi.org/10.1111/j.1365-313X.2006.02997.x">doi:10.1111/j.1365-313X.2006.02997.x</a></p>

<h3 id="may-5">May 5</h3>

<p>Pelsy F, Dumas V, Bévilacqua L, Hocquigny S, Merdinoglu D. Chromosome Replacement and Deletion Lead to Clonal Polymorphism of Berry Color in Grapevine. PLoS Genetics. 2015;11: e1005081. <a href="http://dx.doi.org/10.1371/journal.pgen.1005081">doi:10.1371/journal.pgen.1005081</a></p>

<h3 id="may-4">May 4</h3>

<p>Teng M, Love MI, Davis CA, Djebali S, Dobin A, Graveley BR, et al. A benchmark for RNA-seq quantification pipelines. Genome Biology. 2016;17: 636–12. <a href="http://dx.doi.org/10.1186/s13059-016-0940-1">doi:10.1186/s13059-016-0940-1</a></p>

<h3 id="may-3">May 3</h3>

<p>Zhao J, Akinsanmi I, Arafat D, Cradick TJ, Lee CM, Banskota S, et al. A Burden of Rare Variants Associated with Extremes of Gene Expression in Human Peripheral Blood. The American Journal of Human Genetics. 2016;98: 299–309. <a href="http://dx.doi.org/10.1016/j.ajhg.2015.12.023">doi:10.1016/j.ajhg.2015.12.023</a></p>

<h3 id="may-2">May 2</h3>

<p>Li YI, van de Geijn B, Raj A, Knowles DA, Petti AA, Golan D, et al. RNA splicing is a primary link between genetic variation and disease. Science. American Association for the Advancement of Science; 2016;352: 600–604. <a href="http://dx.doi.org/10.1126/science.aad9417">doi:10.1126/science.aad9417</a></p>

<h3 id="may-1">May 1</h3>

<p>Maumus, F. &amp; Quesneville, H. Ancestral repeats have shaped epigenome and genome composition for millions of years in Arabidopsis thaliana. Nature Communications 5, (2014). <a href="http://dx.doi.org/10.1038/ncomms5104">doi:10.1038/ncomms5104</a></p>

<h3 id="apr-30">Apr 30</h3>

<p>Altman, N. &amp; Krzywinski, M. Points of Significance: Regression diagnostics. Nat Meth 13, 385–386. <a href="http://dx.doi.org/10.1038/nmeth.3854">doi:10.1038/nmeth.3854</a></p>

<h3 id="apr-29">Apr 29</h3>

<p>Baidouri El M, Kim KD, Abernathy B, Arikit S, Maumus F, Panaud O, et al. A new approach for annotation of transposable elements using small RNA mapping. Nucleic Acids Research. 2015;43: e84–e84. <a href="http://dx.doi.org/10.1093/nar/gkv257">doi:10.1093/nar/gkv257</a></p>

<h3 id="apr-28">Apr 28</h3>

<p>Caspi A, Pachter L. Identification of transposable elements using multiple alignments of related genomes. Genome Research. 2006;16: 260–270. <a href="http://dx.doi/10.1101/gr.4361206">doi:10.1101/gr.4361206</a></p>

<h3 id="apr-27">Apr 27</h3>

<p>Flutre T, Permal E, Quesneville H. Transposable element annotation in completely sequenced eukaryote genomes. Plant Transposable Elements. Springer; 2012. pp. 17–39. <a href="http://link.springer.com/chapter/10.1007/978-3-642-31842-9_2">ISBN: 978-3-642-31841-2</a></p>

<h3 id="apr-26">Apr 26</h3>

<p>Exposito-Alonso M, Becker C, Schuenemann VJ, Reitter E, Setzer C, Slovak R, et al. The rate and effect of de novo mutations in natural populations of Arabidopsis thaliana. bioRxiv. 2016. <a href="http://dx.doi.org/10.1101/050203">doi:10.1101/050203</a></p>

<h3 id="apr-25">Apr 25</h3>

<p>Hoen DR, Hickey G, Bourque G, Casacuberta J, Cordaux R, Feschotte C, et al. A call for benchmarking transposable element annotation methods. Mobile DNA. 2015;6: 1–9. <a href="http://dx.doi.org/10.1186/s13100-015-0044-6">doi:10.1186/s13100-015-0044-6</a></p>

<h3 id="apr-24">Apr 24</h3>

<p>Sanjana NE, Wright J, Zheng K, Shalem O, Fontanillas P, Joung J, et al. High-resolution interrogation of functional elements in the noncoding genome. bioRxiv. 2016 Apr p. 049130. <a href="http://dx.doi.org/10.1101/049130">doi:10.1101/049130</a></p>

<h3 id="apr-23">Apr 23</h3>

<p>Conroy RM. What hypotheses do “nonparametric” two-group tests actually test? Stata Journal. College Station, TX: Stata Press; <a href="http://www.stata-journal.com/article.html?article=st0253">2012;12: 182–190.</a></p>

<h3 id="apr-22">Apr 22</h3>

<p>Narasimhan VM, Hunt KA, Mason D, Baker CL, Karczewski KJ, Barnes MR, et al. Health and population effects of rare gene knockouts in adult humans with related parents. Science. 2016;352: 474–477. <a href="http://science.sciencemag.org/content/352/6284/474">doi:10.1126/science.aac8624</a></p>

<h3 id="apr-21">Apr 21</h3>

<p>Bergelson J, Buckler ES, Ecker JR, Nordborg M, Weigel D. A Proposal Regarding Best Practices for Validating the Identity of Genetic Stocks and the Effects of Genetic Variants. The Plant Cell. 2016;28: 606–609. <a href="http://www.plantcell.org/content/28/3/606.long">doi:10.1105/tpc.15.00502</a></p>

<h3 id="apr-20">Apr 20</h3>

<p>Krzywinski M, Altman N. Importance of being uncertain. Nature Methods. 2013;10: 809–810. <a href="http://www.nature.com/nmeth/journal/v10/n9/full/nmeth.2613.html">doi:10.1038/nmeth.2613</a></p>

<h3 id="apr-19">Apr 19</h3>

<p>Zhang Z, Liu X, Guo X, Wang X-J, Zhang X. Arabidopsis AGO3 predominantly recruits 24-nt small RNAs to regulate epigenetic silencing. Nature Plants. <a href="http://www.nature.com/articles/nplants201649">doi:10.1038/nplants.2016.49</a></p>

<h3 id="apr-18">Apr 18</h3>

<p>Diehl WE, Patel N, Halm K, Johnson WE. Tracking interspecies transmission and long-term evolution of an ancient retrovirus using the genomes of modern mammals. Goff SP, editor. eLife. 2016;5: e12704. <a href="http://elifesciences.org/content/5/e12704v1">doi:10.7554/eLife.12704</a></p>

<h3 id="apr-17">Apr 17</h3>

<p>Chen R, Shi L, Hakenberg J, Naughton B, Sklar P, Zhang J, et al. Analysis of 589,306 genomes identifies individuals resilient to severe Mendelian childhood diseases. Nature Biotechnology. <a href="http://www.nature.com/nbt/journal/vaop/ncurrent/full/nbt.3514.html">doi:10.1038/nbt.3514</a></p>

<h3 id="apr-16">Apr 16</h3>

<p>Oikonomopoulos S, Wang YC, Djambazian H, Badescu D, Ragoussis J. Benchmarking of the Oxford Nanopore MinION sequencing for quantitative and qualitative assessment of cDNA populations. bioRxiv. 2016 Apr p. 048074. <a href="http://biorxiv.org/content/early/2016/04/11/048074">doi:10.1101/048074</a></p>

<h3 id="apr-15">Apr 15</h3>

<p>Jeffares DC, Jolly C, Hoti M, Speed D, Rallis C, Dessimoz C, et al. Transient structural variations alter gene expression and quantitative traits in Schizosaccharomyces pombe. bioRxiv. 2016 Apr p. 047266. <a href="http://biorxiv.org/content/biorxiv/early/2016/04/13/047266">doi:10.1101/047266</a></p>

<h3 id="apr-14">Apr 14</h3>

<p>Linheiro RS, Bergman CM. 2012. Whole Genome Resequencing Reveals Natural Target Site Preferences of Transposable Elements in Drosophila melanogaster ed. J.E. Stajich. PLoS ONE 7: e30008. <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0030008">doi: 10.1371/journal.pone.0030008</a></p>

<h3 id="apr-13">Apr 13</h3>

<p>Wilson G. Software Carpentry: lessons learned [version 2; referees: 3 approved]. F1000Research 2016, 3:62 <a href="http://f1000research.com/articles/3-62/v2">doi: 10.12688/f1000research.3-62.v2</a></p>

<h3 id="apr-12">Apr 12</h3>

<p>Lee HC, Lai K, Lorenc MT, Imelfort M, Duran C, Edwards D. 2011. Bioinformatics tools and databases for analysis of next-generation sequence data. Briefings in Functional Genomics. <a href="http://bfg.oxfordjournals.org/content/early/2011/12/18/bfgp.elr037.full">doi:10.1093/bfgp/elr037</a></p>

<h3 id="apr-11">Apr 11</h3>

<p>Muñoz-Lopez M, Garcia-Perez JL. 2010. DNA Transposons: Nature and Applications in Genomics. Curr Genomics 11: 115–128. <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2874221/">doi: 10.2174/138920210790886871</a></p>

<h3 id="apr-10">Apr 10</h3>

<p>Vourekas A, Alexiou P, Vrettos N, Maragkakis M, Mourelatos Z. 2016. Sequence-dependent but not sequence-specific piRNA adhesion traps mRNAs to the germ plasm. Nature 1–24. <a href="http://www.nature.com/nature/journal/v531/n7594/full/nature17150.html">doi:10.1038/nature17150</a></p>

<h3 id="apr-9">Apr 9</h3>

<p>Kanno S, Arrighi J-F, Chiarenza S, Bayle V, Berthomé R, Péret B, Javot H, Delannoy E, Marin E, Nakanishi TM, et al. 2016. A novel role for the root cap in phosphate uptake and homeostasis ed. D. Weigel. eLife 5: e14577. <a href="http://elifesciences.org/content/5/e14577v1">doi:10.7554/eLife.14577</a></p>

<h3 id="apr-8">Apr 8</h3>

<p>Calderwood A, Kopriva S, Morris RJ. 2016. Transcript Abundance Explains mRNA Mobility Data in Arabidopsis thaliana. The Plant Cell 28: 610–615. <a href="http://www.plantcell.org/content/28/3/610">doi:10.1105/tpc.15.00956</a></p>

<h3 id="apr-7">Apr 7</h3>

<p>Bewick AJ, Ji L, Niederhuth CE, Willing E-M, Hofmeister BT, Shi X, Wang L, Lu Z, Rohr NA, Hartwig B, et al. 2016. On the Origin and Evolutionary Consequences of Gene Body DNA Methylation. bioRxiv. <a href="http://biorxiv.org/content/biorxiv/early/2016/03/24/045542">doi:10.1101/045542</a></p>

<h3 id="apr-6">Apr 6</h3>

<p>Simpson JT, Workman R, Zuzarte PC, David M, Dursi LJ, Timp W. 2016. Detecting DNA Methylation using the Oxford Nanopore Technologies MinION sequencer. bioRxiv. <a href="http://biorxiv.org/content/biorxiv/early/2016/04/04/047142">doi: 10.1101/047142</a></p>

<p>Rand AC, Jain M, Eizenga J, Musselman-Brown A, Olsen HE, Akeson M, Paten B. 2016. Cytosine Variant Calling with High-throughput Nanopore Sequencing. bioRxiv <a href="http://biorxiv.org/content/biorxiv/early/2016/04/04/047134">doi:10.1101/047134</a></p>

<h3 id="apr-5">Apr 5</h3>

<p>Toomajian C, Ajioka RS, Jorde LB, Kushner JP, Kreitman M. 2003. A method for detecting recent selection in the human genome from allele age estimates. Genetics 165: 287–297. <a href="http://www.ncbi.nlm.nih.gov/pubmed/14504236">PMC1462736</a></p>

<h3 id="apr-4">Apr 4</h3>

<p>Genin E, Tullio-Pelet A, Begeot F, Lyonnet S, Abel L. 2004. Estimating the age of rare disease mutations: the example of Triple-A syndrome. Journal of Medical Genetics 41: 445–449. <a href="http://jmg.bmj.com/content/41/6/445.full">doi:10.1136/jmg.2003.017962</a></p>

<h3 id="apr-3">Apr 3</h3>

<p>Slatkin M. 2008. Linkage disequilibrium — understanding the evolutionary past and mapping the medical future. Nature Reviews Genetics 9: 477–485. <a href="http://www.nature.com/nrg/journal/v9/n6/full/nrg2361.html">doi:10.1038/nrg2361</a></p>

<h3 id="apr-2">Apr 2</h3>

<p>Ben-Moshe S, Itzkovitz S. 2016. Bursting through the cell cycle. eLife 5: e14953. <a href="http://elifesciences.org/content/5/e14953v1">doi:10.7554/eLife.14953</a></p>

<h3 id="apr-1">Apr 1</h3>

<p>Bourne HR. 2013. The writing on the wall. eLife 2: e00642. <a href="http://elifesciences.org/content/2/e00642v1">doi: 10.7554/eLife.00642</a></p>

<h3 id="mar-31">Mar 31</h3>

<p>McClintock B. 1950. The origin and behavior of mutable loci in maize. Proc Natl Acad Sci USA 36: 344–355. <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1063197/">PMC1063197</a></p>

<h3 id="mar-30">Mar 30</h3>

<p>Berry S, Hartley M, Olsson TSG, Dean C, Howard M. 2015. Local chromatin environment of a Polycomb target gene instructs its own epigenetic inheritance ed. D. Zilberman. eLife 4: e07205. <a href="http://elifesciences.org/content/4/e07205v2">doi:10.7554/eLife.07205</a></p>

<h3 id="mar-29">Mar 29</h3>

<p>Sánchez-Gracia A, Maside X, Charlesworth B. High rate of horizontal transfer of transposable elements in Drosophila. Trends in Genetics 21: 200–203. <a href="http://www.sciencedirect.com/science/article/pii/S0168952505000417">doi:10.1016/j.tig.2005.02.001</a></p>

<h3 id="mar-28">Mar 28</h3>

<p>Hill T, Schlötterer C, Betancourt AJ. 2016. Hybrid Dysgenesis in Drosophila simulans Associated with a Rapid Invasion of the P-Element ed. H.S. Malik. PLoS Genetics 12: e1005920. <a href="http://journals.plos.org/plosgenetics/article?id=10.1371%2Fjournal.pgen.1005920&amp;utm_source=feedburner&amp;utm_medium=feed&amp;utm_campaign=Feed%3A+plosgenetics%2FNewArticles+%28PLOS+Genetics+-+New+Articles%29">doi: 10.1371/journal.pgen.1005920</a></p>

<h3 id="mar-27">Mar 27</h3>

<p>Greaves IK, Groszmann M, Ying H, Taylor JM, Peacock WJ, Dennis ES. 2012. Trans Chromosomal Methylation in Arabidopsis hybrids. Proc Natl Acad Sci USA 109: 3570–3575. <a href="http://www.pnas.org/content/109/9/3570.full">doi:10.1073/pnas.1201043109</a></p>

<h3 id="mar-26">Mar 26</h3>

<p>Rigal M, Becker C, Pélissier T, Pogorelcnik R, Devos J, Ikeda Y, Weigel D, Mathieu O. 2016. Epigenome confrontation triggers immediate reprogramming of DNA methylation and transposon silencing in Arabidopsis thaliana F1 epihybrids. Proceedings of the National Academy of Sciences 201600672–10. <a href="http://www.pnas.org/content/early/2016/03/16/1600672113.full">doi: 10.1073/pnas.1600672113</a></p>

<h3 id="mar-25">Mar 25</h3>

<p>Groszmann M, Greaves IK, Albertyn ZI, Scofield GN, Peacock WJ, Dennis ES. 2011. Changes in 24-nt siRNA levels in Arabidopsis hybrids suggest an epigenetic contribution to hybrid vigor. Proceedings of the National Academy of Sciences 108: 2617–2622. <a href="http://www.pnas.org/content/108/6/2617.full">doi: 10.1073/pnas.1019217108</a></p>

<h3 id="mar-24">Mar 24</h3>

<p>Fang X, Qi Y. 2016. RNAi in Plants: An Argonaute-Centered View. Plant Cell. Advance Publication February 11, 2016. <a href="http://www.plantcell.org/content/early/2016/02/11/tpc.15.00920.abstract">doi: 10.​1105/​tpc.​15.​00920</a></p>

<h3 id="mar-23">Mar 23</h3>

<p>Miga KH, Eisenhart C, Kent WJ. 2015. Utilizing mapping targets of sequences underrepresented in the reference assembly to reduce false positive alignments. Nucleic Acids Research 43: e133–8. <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4787761/">doi: 10.1093/nar/gkv671</a></p>

<h3 id="mar-22">Mar 22</h3>

<p>Martínez G, Panda K, Köhler C, Slotkin RK. 2016. Silencing in sperm cells is directed by RNA movement from the surrounding nurse cell. Nature Plants 1–8. <a href="http://www.nature.com/articles/nplants201630">DOI: 10.1038/NPLANTS.2016.30</a></p>

<h3 id="mar-21">Mar 21</h3>

<p>Yamamuro C, Miki D, Zheng Z, Ma J, Wang J, Yang Z, Dong J, Zhu J-K. 1AD. Overproduction of stomatal lineage cells in Arabidopsis mutants defective in active DNA demethylation. Nature Communications 5: 1–7. <a href="http://www.nature.com/ncomms/2014/140605/ncomms5062/full/ncomms5062.html">DOI: 10.1038/ncomms5062</a></p>

<h3 id="mar-20">Mar 20</h3>

<p>Saito Y, Tsuji J, Mituyama T. 2014. Bisulfighter: accurate detection of methylated cytosines and differentially methylated regions. Nucleic Acids Research. <a href="http://www.ncbi.nlm.nih.gov/pubmed/24423865">doi:10.1093/nar/gkt1373</a></p>

<h3 id="mar-19">Mar 19</h3>

<p>Guo W, Fiziev P, Yan W, Cokus S, Sun X, Zhang MQ, Chen P-Y, Pellegrini M. 2013. BS-Seeker2: a versatile aligning pipeline for bisulfite sequencing data. BMC Genomics 14: 774. <a href="http://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-14-774">DOI: 10.1186/1471-2164-14-774</a></p>

<h3 id="mar-18">Mar 18</h3>

<p>Elvira-Matelot E, Hachet M, Shamandi N, Comella P, Saez-Vásquez J, Zytnicki M, Vaucheret H. 2016. Arabidopsis RNASE THREE LIKE2 Modulates the Expression of Protein-Coding Genes via 24-Nucleotide Small Interfering RNA-Directed DNA Methylation. The Plant Cell 28: 406–425. <a href="http://www.plantcell.org/content/28/2/406.short?rss=1">doi:10.1105/tpc.15.00540</a></p>

<h3 id="mar-17">Mar 17</h3>

<p>Adrian J, Chang J, Ballenger CE, Bargmann BOR, Alassimone J, Davies KA, Lau OS, Matos JL, Hachez C, Lanctot A, et al. 2015. Transcriptome Dynamics of the Stomatal Lineage: Birth, Amplification, and Termination of a Self- Renewing Population. Dev Cell 33: 107–118. <a href="http://www.sciencedirect.com/science/article/pii/S1534580715000684">doi:10.1016/j.devcel.2015.01.025</a></p>

<h3 id="mar-16">Mar 16</h3>

<p>Maumus F, Quesneville H. 2016. Impact and insights from ancient repetitive elements in plant genomes. Current Opinion in Plant Biology 30: 41–46. <a href="http://www.sciencedirect.com/science/article/pii/S1369526616300036">doi:10.1016/j.pbi.2016.01.003</a></p>

<h3 id="mar-15">Mar 15</h3>

<p>Grandbastien MA, Audeon C, Bonnivard E, Casacuberta JM, Chalhoub B, Costa APP, Le QH, Melayah D, Petit M, Poncet C, et al. 2005. Stress activation and genomic impact of Tnt1 retrotransposons in Solanaceae. Cytogenet Genome Res 110: 229–241. <a href="http://www.ncbi.nlm.nih.gov/pubmed/16093677">DOI: 10.1159/000084957</a></p>

<h3 id="mar-14">Mar 14</h3>

<p>Wolff P, Jiang H, Wang G, Santos-González J, Köhler C. 2015. Paternally expressed imprinted genes establish postzygotic hybridization barriers in Arabidopsis thaliana. eLife. <a href="http://elifesciences.org/content/4/e10074v2#api_box">DOI: 10.7554/eLife.10074</a></p>

<h3 id="mar-13">Mar 13</h3>

<p>Weiß CL, Dannemann M, Prüfer K, Burbano HA. 2015. Contesting the presence of wheat in the British Isles 8,000 years ago by assessing ancient DNA authenticity from low-coverage data ed. J.K. Pickrell. eLife 4: e10005. <a href="http://elifesciences.org/content/4/e10005v1">DOI: 10.7554/eLife.10005</a></p>

<h3 id="mar-12">Mar 12</h3>

<p>Loose M, Malla S, Stout M. 2016. Real time selective sequencing using nanopore technology. bioRxiv 038760. <a href="http://biorxiv.org/content/early/2016/02/03/038760">doi: 10.1101/038760</a></p>

<h3 id="mar-11">Mar 11</h3>

<p>Mita P, Boeke JD. 2016. How retrotransposons shape genome regulation. Current Opinion in Genetics &amp; Development 37: 90–100. <a href="http://www.sciencedirect.com/science/article/pii/S0959437X16000022">doi:10.1016/j.gde.2016.01.001</a></p>

<h3 id="mar-10">Mar 10</h3>

<p>O’Neill RJW, O’Neill MJ, Graves JAM. 1998. Undermethylation associated with retroelement activation and chromosome remodelling in an interspecific mammalian hybrid. Nature 393: 68–72. <a href="http://www.nature.com/doifinder/10.1038/29985">doi:10.1038/29985</a></p>

<h3 id="mar-9">Mar 9</h3>

<p>Zeh DW, Zeh JA, Ishida Y. 2009. Transposable elements and an epigenetic basis for punctuated equilibria. BioEssays 31: 715–726. <a href="http://onlinelibrary.wiley.com/enhanced/doi/10.1002/bies.200900026">doi:10.1002/bies.200900026</a></p>

<h3 id="mar-8">Mar 8</h3>

<p>Rosenthal R, Fode KL. 2007. The effect of experimenter bias on the performance of the albino rat. Syst Res 8: 183–189. <a href="http://onlinelibrary.wiley.com/doi/10.1002/bs.3830080302/abstract">doi:10.1002/bs.3830080302</a></p>

<h3 id="mar-7">Mar 7</h3>

<p>Rivas MA, Graham D, Sulem P, Stevens C, Desch N. 2015. A protein truncating R179X variant in RNF186 confers protection against ulcerative colitis. bioRxiv. <a href="http://dx.doi.org/10.1101/035105">doi:10.1101/035105</a></p>

<h3 id="mar-6">Mar 6</h3>

<p>Yuan K, O’Farrell PH. 2016. TALE-light imaging reveals maternally guided, H3K9me2/3-independent emergence of functional heterochromatin in Drosophila embryos. Genes &amp; Development 30: 579–593. <a href="http://genesdev.cshlp.org/content/30/5/579.short?rss=1">doi:10.1101/gad.272237.115.</a></p>

<h3 id="mar-5">Mar 5</h3>

<p>Sul JH, Bilow M, Yang W-Y, Kostem E, Furlotte N, He D, Eskin E. 2016. Accounting for Population Structure in Gene-by-Environment Interactions in Genome-Wide Association Studies Using Mixed Models ed. N.J. Schork. PLoS Genetics 12: e1005849–19. <a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1005849">DOI: 10.1371/journal.pgen.1005849</a></p>

<h3 id="mar-4">Mar 4</h3>

<p>Hsi-Yang Fritz M, Leinonen R, Cochrane G, Birney E. 2011. Efficient storage of high throughput DNA sequencing data using reference-based compression. Genome Research 21: 734–740. <a href="http://genome.cshlp.org/content/21/5/734.long">doi:10.1101/gr.114819.110.</a></p>

<h3 id="mar-3">Mar 3</h3>

<p>Levasseur A, Bekliz M, Chabrière E, Pontarotti P, La Scola B, Raoult D. 2016. MIMIVIRE is a defence system in mimivirus that confers resistance to virophage. Nature 1–12. <a href="http://www.nature.com/nature/journal/vaop/ncurrent/full/nature17146.html">doi:10.1038/nature17146</a></p>

<h3 id="mar-2">Mar 2</h3>

<p>Bogdanovic O, Smits AH, la Calle Mustienes de E, Tena JJ, Ford E, Williams R, Senanayake U, Schultz MD, Hontelez S, van Kruijsbergen I, et al. 2016. Active DNA demethylation at enhancers during the vertebrate phylotypic period. Nat Genet 1–13. <a href="http://dx.doi.org/10.1038/ng.3522">doi:10.1038/ng.3522</a></p>

<h3 id="mar-1">Mar 1</h3>

<p>Lewsey MG, Hardcastle TJ, Melnyk CW, Molnar A, Valli A, Urich MA, Nery JR, Baulcombe DC, Ecker JR. 2016. Mobile small RNAs regulate genome-wide DNA methylation. Proceedings of the National Academy of Sciences. <a href="http://www.pnas.org/content/113/6/E801.abstract">doi:10.1073/pnas.1515072113</a></p>

<h3 id="feb-29">Feb 29</h3>

<p>Storey JD, Tibshirani R. 2003. Statistical significance for genomewide studies. Proc Natl Acad Sci USA 100: 9440–9445. <a href="http://www.pnas.org/content/100/16/9440.full">doi:10.1073/pnas.1530509100</a></p>

<h3 id="feb-28">Feb 28</h3>

<p>Fernandez AG, Mis EK, Bargmann BOR, Birnbaum KD, Piano F. 2010. Automated sorting of live C. elegans using laFACS. Nat Meth 7: 417–418. <a href="http://www.nature.com/nmeth/journal/v7/n6/full/nmeth.f.304.html">doi:10.1038/nmeth.f.304.</a></p>

<h3 id="feb-27">Feb 27</h3>

<p>Slaughter A, Daniel X, Flors V, Luna E, Hohn B, Mauch-Mani B. 2012. Descendants of Primed Arabidopsis Plants Exhibit Resistance to Biotic Stress. Plant Physiology 158: 835–843. <a href="http:/​/​dx.​doi.​org/​10.​1104/​pp.​111.​191593">doi:​10.​1104/​pp.​111.​191593</a></p>

<h3 id="feb-26">Feb 26</h3>

<p>Guerreiro MPG. 2014. Interspecific hybridization as a genomic stressor inducing mobilization of transposable elements in Drosophila. Mob Genet Elements 4: e34394. <a href="http://dx.doi.org/10.4161/mge.34394">doi.org/10.4161/mge.34394</a></p>

<h3 id="feb-25">Feb 25</h3>

<p>Vale RD. 2015. Accelerating Scientific Publication in Biology. bioRxiv 022368. <a href="http://biorxiv.org/content/early/2015/09/12/022368">doi:10.1101/022368</a></p>

<blockquote>
  <p><em>Related:</em>
<a href="http://www.michaeleisen.org/blog/?p=1733">http://www.michaeleisen.org/blog/?p=1733</a></p>
</blockquote>

<h3 id="feb-24">Feb 24</h3>

<p>Clavel M, Pélissier T, Descombin J, Jean V, Picart C, Charbonel C, Saez-Vásquez J, Bousquet-Antonelli C, Deragon J-M. 2015. Parallel action of AtDRB2 and RdDM in the control of transposable element expression. BMC Plant Biology 15: 70. <a href="http://bmcplantbiol.biomedcentral.com/articles/10.1186/s12870-015-0455-z">DOI 10.1186/s12870-015-0455-z</a></p>

<h3 id="feb-23">Feb 23</h3>

<p>Crisp PA, Ganguly D, Eichten SR, Borevitz JO, Pogson BJ. 2016. Reconsidering plant memory: Intersections between stress recovery, RNA turnover, and epigenetics. Science Advances 2: e1501340–e1501340. <a href="http://advances.sciencemag.org/content/advances/2/2/e1501340.full">doi:10.1126/sciadv.1501340</a></p>

<p><em>Interesting or important points:</em></p>

<blockquote>
  <p>An as yet entirely unexplored possibility is the prospect that transcriptional memory might be underpinned by changes in mRNA stability rather than transcription.</p>
</blockquote>

<p>I think certain RNA modifications have been shown to be induced by stress (at least in yeast), perhaps they play a role in modifying mRNA stability to facilitate stress memory.</p>

</div>


      <!-- <div class="footer">
        <p>
          &copy; 2016. All rights reserved.
        </p>
      </div> -->
    </div>

  </body>
</html>
