<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">

  <head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  <!-- Enable responsiveness on mobile devices-->
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>
    
      Tim Stuart &middot; 
    
  </title>

  <!-- CSS -->
  <link rel="stylesheet" href="/public/css/poole.css">
  <link rel="stylesheet" href="/public/css/syntax.css">

  <!-- Icons -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/public/apple-touch-icon-precomposed.png">
  <link rel="shortcut icon" href="/public/favicon.ico">

  <!-- RSS -->
  <link rel="alternate" type="application/rss+xml" title="RSS" href="/atom.xml">
</head>

  <script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-54133406-1', 'auto');
  ga('send', 'pageview');

</script>


  <body>

    <div class="container content">
      <div class="masthead">
        <h3 class="masthead-title">
          <a href="/" title="Home">Tim Stuart</a>
          <small></small>

          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/about">About</a></small>
          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/archive">Archive</a></small>
          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/code">Code</a></small>
          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/publications">Publications</a></small>
          
            &nbsp;&nbsp;&nbsp;
            <small><a href="/atom.xml">Feed</a></small>
          
        </h3>
      </div>

      <div class="posts">
  
  <div class="post">
    <h1 class="post-title">
      <a href="/2014/08/25/UsefulPython/">
        Useful python code
      </a>
    </h1>

    <span class="post-date">25 Aug 2014</span>

    <h3 id="take-command-line-arguments">Take command-line arguments</h3>

<p>```python
from argparse import ArgumentParser</p>

<p>version = pkg_resources.require(“program”)[0].version
parser = ArgumentParser(description=’program description’)
group = parser.add_mutually_exclusive_group()
parser.add_argument(‘–version’, action=’version’, version=’%(prog)s ‘+str(version))
parser.add_argument(‘–option’, help=’option description’, required=False, default=False, action=’store_true’)
parser.add_argument(‘-n’, ‘–name’, help=’sample name’, required=True)
```</p>

<h2 id="lists">Lists</h2>

<h3 id="list-comprehension">List comprehension</h3>

<p>If ever declaring and empty list then using <code>.append</code>, can be done with list comprehension.</p>

<p>```python
list3 = []
for x in list2:
    if x[0] in list1:
        list3.append(x)</p>

<p>list3 = [x for x in list2 if x[0] in list1]
```</p>

<p>Can also use this to read from a file in one line, eg to read column 4 in a file into a list in python:</p>

<p><code>python
all_TEs = [line.rsplit()[4] for line in open("TAIR9_TE.bed", "r")]
</code></p>

<p>Or read all lines into a list:</p>

<p><code>python
all_lines = [line.strip("\n") for line in open("filename", "r")]
</code></p>

<h3 id="join-items-in-list">Join items in list</h3>

<p><code>python
outfile.write('{l}\n'.format(l='\t'.join(list)))
</code></p>

<p>If list items are not strings</p>

<p><code>python
outfile.write('{l}\n'.format(l=map(str, list)))
</code></p>

<p>Or using list comprehension</p>

<p><code>python
stringVersion = [str(x) for x in inputList]
</code></p>

<h2 id="printing">Printing</h2>

<h3 id="print-in-colour">Print in colour</h3>

<p>```python
class colour:
    PURPLE = ‘\033[95m’
    CYAN = ‘\033[96m’
    DARKCYAN = ‘\033[36m’
    BLUE = ‘\033[94m’
    GREEN = ‘\033[92m’
    YELLOW = ‘\033[93m’
    RED = ‘\033[91m’
    BOLD = ‘\033[1m’
    UNDERLINE = ‘\033[4m’
    END = ‘\033[0m’</p>

<p>print colour.CYAN + ‘This is blue’ + colour.END
```</p>

<h3 id="format-text-wrapping-correctly-in-terminal">Format text wrapping correctly in terminal</h3>

<p><code>python
import textwrap
print textwrap.dedent('print message')
</code></p>

<h2 id="strings">Strings</h2>

<h3 id="split-strings-by-multiple-delimiters">Split strings by multiple delimiters</h3>

<p><code>python
accession_name = f.replace('calls_', '.').split('.')
</code></p>

<h2 id="file-reading--writing">File reading / writing</h2>

<h3 id="read-lines-in-tsvcsv-file">Read lines in tsv/csv file</h3>

<p><code>python
with open('merged_dmrs.bed', 'r') as f:
    for line in f:
        line = line.rsplit() # each item in line is now part of a list
</code></p>

<h2 id="find-if-coordinates-overlap">Find if coordinates overlap</h2>

<p><code>python
def overlap(start1, stop1, start2, stop2):
    """returns True if sets of coordinates overlap. Assumes coordinates are on same chromosome"""
    for y in xrange(start2, stop2+1):
        if start1 &lt;= y &lt;= stop1:
            return True
        else:
            pass
</code></p>

<h2 id="functions-for-mapping-ngs-reads">Functions for mapping NGS reads</h2>

<p>```python
import os
from subprocess import call</p>

<p>def peMap(proc):
    “””
    paired-end reads
    “””
    for filename in os.listdir(‘.’):
        if filename.endswith(“<em>1.fastq”):
            sam = filename.split(‘</em>’)
            sam = sam[0]
            call([“bowtie2”, “-p{x}”.format(x=proc), “-q”, “-x /dd_stage/userdata/lister/data/genomes/bowtie2_indexes/tair9”,
                  “-1 {f}”.format(f=sam + ‘_1.fastq’), “-2 {f}”.format(f=sam + ‘_2.fastq’), “-S {s}”.format(s=sam + ‘.sam’)])
        else:
            pass</p>

<p>def seMap(proc):
    “””
    single-end reads
    “””
    for filename in os.listdir(‘.’):
        if filename.endswith(“.fastq”):
            sam = filename.split(‘.’)
            sam = sam[0]
            call([“bowtie2”, “-p{x}”.format(x=proc), “-q”, “-x /dd_stage/userdata/lister/data/genomes/bowtie2_indexes/tair9”,
                  “-U {f}”.format(f=filename), “-S {s}”.format(s=sam + ‘.sam’)])
        else:
            pass
```</p>

<h2 id="split-sra-files-into-fastq">Split sra files into fastq</h2>

<p>```python
import os
from subprocess import call</p>

<p>def fastqSplit():
    for filename in os.listdir(‘.’):
        if filename.endswith(‘.sra’):
            print ‘processing {n}’.format(n=filename)
            call([‘fastq-dump’, ‘–split-3’, ‘-v’, filename])
        else:
            pass
```</p>

  </div>
  
</div>

<div class="pagination">
  
    <a class="pagination-item older" href="/page9">Older</a>
  
  
    
      <a class="pagination-item newer" href="/page7">Newer</a>
    
  
</div>


      <!-- <div class="footer">
        <p>
          &copy; 2016. All rights reserved.
        </p>
      </div> -->
    </div>

  </body>
</html>
